{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ClinVar\n",
    "\n",
    "Original data was pulled from [here](https://ftp.ncbi.nlm.nih.gov/pub/clinvar/tab_delimited/variant_summary.txt.gz) (FTP link) on 2/14/24."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import table_cleaning_functions as tcf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Preliminary Wrangling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_5369/4165182006.py:2: DtypeWarning: Columns (18) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  clinvar_raw = pd.read_csv('../home/data/raw_data/GlyGen/clinvar_raw.txt', sep = '\\t')\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(4807886, 40)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#AlleleID</th>\n",
       "      <th>Type</th>\n",
       "      <th>Name</th>\n",
       "      <th>GeneID</th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>ClinicalSignificance</th>\n",
       "      <th>ClinSigSimple</th>\n",
       "      <th>LastEvaluated</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>...</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PositionVCF</th>\n",
       "      <th>ReferenceAlleleVCF</th>\n",
       "      <th>AlternateAlleleVCF</th>\n",
       "      <th>SomaticClinicalImpact</th>\n",
       "      <th>SomaticClinicalImpactLastEvaluated</th>\n",
       "      <th>ReviewStatusClinicalImpact</th>\n",
       "      <th>Oncogenicity</th>\n",
       "      <th>OncogenicityLastEvaluated</th>\n",
       "      <th>ReviewStatusOncogenicity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15041</td>\n",
       "      <td>Indel</td>\n",
       "      <td>NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTA...</td>\n",
       "      <td>9907</td>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>HGNC:22197</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>397704705</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>4820844</td>\n",
       "      <td>GGAT</td>\n",
       "      <td>TGCTGTAAACTGTAACTGTAAA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>15041</td>\n",
       "      <td>Indel</td>\n",
       "      <td>NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTA...</td>\n",
       "      <td>9907</td>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>HGNC:22197</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>0</td>\n",
       "      <td>-</td>\n",
       "      <td>397704705</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>4781213</td>\n",
       "      <td>GGAT</td>\n",
       "      <td>TGCTGTAAACTGTAACTGTAAA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15042</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>NM_014855.3(AP5Z1):c.1413_1426del (p.Leu473fs)</td>\n",
       "      <td>9907</td>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>HGNC:22197</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>0</td>\n",
       "      <td>Jun 29, 2010</td>\n",
       "      <td>397704709</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>4827360</td>\n",
       "      <td>GCTGCTGGACCTGCC</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>15042</td>\n",
       "      <td>Deletion</td>\n",
       "      <td>NM_014855.3(AP5Z1):c.1413_1426del (p.Leu473fs)</td>\n",
       "      <td>9907</td>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>HGNC:22197</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>0</td>\n",
       "      <td>Jun 29, 2010</td>\n",
       "      <td>397704709</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>4787729</td>\n",
       "      <td>GCTGCTGGACCTGCC</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15043</td>\n",
       "      <td>single nucleotide variant</td>\n",
       "      <td>NM_014630.3(ZNF592):c.3136G&gt;A (p.Gly1046Arg)</td>\n",
       "      <td>9640</td>\n",
       "      <td>ZNF592</td>\n",
       "      <td>HGNC:28986</td>\n",
       "      <td>Uncertain significance</td>\n",
       "      <td>0</td>\n",
       "      <td>Jun 29, 2015</td>\n",
       "      <td>150829393</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>85342440</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 40 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   #AlleleID                       Type  \\\n",
       "0      15041                      Indel   \n",
       "1      15041                      Indel   \n",
       "2      15042                   Deletion   \n",
       "3      15042                   Deletion   \n",
       "4      15043  single nucleotide variant   \n",
       "\n",
       "                                                Name  GeneID GeneSymbol  \\\n",
       "0  NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTA...    9907      AP5Z1   \n",
       "1  NM_014855.3(AP5Z1):c.80_83delinsTGCTGTAAACTGTA...    9907      AP5Z1   \n",
       "2     NM_014855.3(AP5Z1):c.1413_1426del (p.Leu473fs)    9907      AP5Z1   \n",
       "3     NM_014855.3(AP5Z1):c.1413_1426del (p.Leu473fs)    9907      AP5Z1   \n",
       "4       NM_014630.3(ZNF592):c.3136G>A (p.Gly1046Arg)    9640     ZNF592   \n",
       "\n",
       "      HGNC_ID    ClinicalSignificance  ClinSigSimple LastEvaluated  \\\n",
       "0  HGNC:22197              Pathogenic              0             -   \n",
       "1  HGNC:22197              Pathogenic              0             -   \n",
       "2  HGNC:22197              Pathogenic              0  Jun 29, 2010   \n",
       "3  HGNC:22197              Pathogenic              0  Jun 29, 2010   \n",
       "4  HGNC:28986  Uncertain significance              0  Jun 29, 2015   \n",
       "\n",
       "   RS# (dbSNP)  ... VariationID PositionVCF ReferenceAlleleVCF  \\\n",
       "0    397704705  ...           2     4820844               GGAT   \n",
       "1    397704705  ...           2     4781213               GGAT   \n",
       "2    397704709  ...           3     4827360    GCTGCTGGACCTGCC   \n",
       "3    397704709  ...           3     4787729    GCTGCTGGACCTGCC   \n",
       "4    150829393  ...           4    85342440                  G   \n",
       "\n",
       "       AlternateAlleleVCF SomaticClinicalImpact  \\\n",
       "0  TGCTGTAAACTGTAACTGTAAA                     -   \n",
       "1  TGCTGTAAACTGTAACTGTAAA                     -   \n",
       "2                       G                     -   \n",
       "3                       G                     -   \n",
       "4                       A                     -   \n",
       "\n",
       "  SomaticClinicalImpactLastEvaluated ReviewStatusClinicalImpact Oncogenicity  \\\n",
       "0                                  -                          -            -   \n",
       "1                                  -                          -            -   \n",
       "2                                  -                          -            -   \n",
       "3                                  -                          -            -   \n",
       "4                                  -                          -            -   \n",
       "\n",
       "  OncogenicityLastEvaluated  ReviewStatusOncogenicity  \n",
       "0                         -                         -  \n",
       "1                         -                         -  \n",
       "2                         -                         -  \n",
       "3                         -                         -  \n",
       "4                         -                         -  \n",
       "\n",
       "[5 rows x 40 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read in raw clinvar data \n",
    "clinvar_raw = pd.read_csv('../home/data/raw_data/GlyGen/clinvar_raw.txt', sep = '\\t')\n",
    "print(clinvar_raw.shape)\n",
    "clinvar_raw.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "#AlleleID                             0\n",
       "Type                                  0\n",
       "Name                                  0\n",
       "GeneID                                0\n",
       "GeneSymbol                            0\n",
       "HGNC_ID                               0\n",
       "ClinicalSignificance                  0\n",
       "ClinSigSimple                         0\n",
       "LastEvaluated                         0\n",
       "RS# (dbSNP)                           0\n",
       "nsv/esv (dbVar)                       0\n",
       "RCVaccession                          0\n",
       "PhenotypeIDS                          0\n",
       "PhenotypeList                         0\n",
       "Origin                                0\n",
       "OriginSimple                          0\n",
       "Assembly                              0\n",
       "ChromosomeAccession                   0\n",
       "Chromosome                            0\n",
       "Start                                 0\n",
       "Stop                                  0\n",
       "ReferenceAllele                       0\n",
       "AlternateAllele                       0\n",
       "Cytogenetic                           0\n",
       "ReviewStatus                          0\n",
       "NumberSubmitters                      0\n",
       "Guidelines                            0\n",
       "TestedInGTR                           0\n",
       "OtherIDs                              0\n",
       "SubmitterCategories                   0\n",
       "VariationID                           0\n",
       "PositionVCF                           0\n",
       "ReferenceAlleleVCF                    0\n",
       "AlternateAlleleVCF                    0\n",
       "SomaticClinicalImpact                 0\n",
       "SomaticClinicalImpactLastEvaluated    0\n",
       "ReviewStatusClinicalImpact            0\n",
       "Oncogenicity                          0\n",
       "OncogenicityLastEvaluated             0\n",
       "ReviewStatusOncogenicity              0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_raw.isnull().sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Drop unnecessary rows."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['#AlleleID', 'Type', 'Name', 'GeneID', 'GeneSymbol', 'HGNC_ID',\n",
       "       'ClinicalSignificance', 'ClinSigSimple', 'LastEvaluated', 'RS# (dbSNP)',\n",
       "       'nsv/esv (dbVar)', 'RCVaccession', 'PhenotypeIDS', 'PhenotypeList',\n",
       "       'Origin', 'OriginSimple', 'Assembly', 'ChromosomeAccession',\n",
       "       'Chromosome', 'Start', 'Stop', 'ReferenceAllele', 'AlternateAllele',\n",
       "       'Cytogenetic', 'ReviewStatus', 'NumberSubmitters', 'Guidelines',\n",
       "       'TestedInGTR', 'OtherIDs', 'SubmitterCategories', 'VariationID',\n",
       "       'PositionVCF', 'ReferenceAlleleVCF', 'AlternateAlleleVCF',\n",
       "       'SomaticClinicalImpact', 'SomaticClinicalImpactLastEvaluated',\n",
       "       'ReviewStatusClinicalImpact', 'Oncogenicity',\n",
       "       'OncogenicityLastEvaluated', 'ReviewStatusOncogenicity'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_raw.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['#AlleleID',\n",
       " 'Type',\n",
       " 'Name',\n",
       " 'GeneID',\n",
       " 'GeneSymbol',\n",
       " 'HGNC_ID',\n",
       " 'ClinicalSignificance',\n",
       " 'ClinSigSimple',\n",
       " 'LastEvaluated',\n",
       " 'RS# (dbSNP)',\n",
       " 'nsv/esv (dbVar)',\n",
       " 'RCVaccession',\n",
       " 'PhenotypeIDS',\n",
       " 'PhenotypeList',\n",
       " 'Origin',\n",
       " 'OriginSimple',\n",
       " 'Assembly',\n",
       " 'ChromosomeAccession',\n",
       " 'Chromosome',\n",
       " 'Start',\n",
       " 'Stop',\n",
       " 'ReferenceAllele',\n",
       " 'AlternateAllele',\n",
       " 'Cytogenetic',\n",
       " 'ReviewStatus',\n",
       " 'NumberSubmitters',\n",
       " 'Guidelines',\n",
       " 'TestedInGTR',\n",
       " 'OtherIDs',\n",
       " 'SubmitterCategories',\n",
       " 'VariationID',\n",
       " 'PositionVCF',\n",
       " 'ReferenceAlleleVCF',\n",
       " 'AlternateAlleleVCF',\n",
       " 'SomaticClinicalImpact',\n",
       " 'SomaticClinicalImpactLastEvaluated',\n",
       " 'ReviewStatusClinicalImpact',\n",
       " 'Oncogenicity',\n",
       " 'OncogenicityLastEvaluated',\n",
       " 'ReviewStatusOncogenicity']"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = list(clinvar_raw.columns)\n",
    "cols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>PhenotypeList</th>\n",
       "      <th>VariationID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>397704705</td>\n",
       "      <td>Hereditary spastic paraplegia 48</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>397704705</td>\n",
       "      <td>Hereditary spastic paraplegia 48</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>397704709</td>\n",
       "      <td>Hereditary spastic paraplegia 48</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AP5Z1</td>\n",
       "      <td>397704709</td>\n",
       "      <td>Hereditary spastic paraplegia 48</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ZNF592</td>\n",
       "      <td>150829393</td>\n",
       "      <td>Galloway-Mowat syndrome 1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  GeneSymbol  RS# (dbSNP)                     PhenotypeList  VariationID\n",
       "0      AP5Z1    397704705  Hereditary spastic paraplegia 48            2\n",
       "1      AP5Z1    397704705  Hereditary spastic paraplegia 48            2\n",
       "2      AP5Z1    397704709  Hereditary spastic paraplegia 48            3\n",
       "3      AP5Z1    397704709  Hereditary spastic paraplegia 48            3\n",
       "4     ZNF592    150829393         Galloway-Mowat syndrome 1            4"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drop_cols = [x for x in cols if x not in {'GeneSymbol', 'RS# (dbSNP)', 'PhenotypeList', 'VariationID'}]\n",
    "clinvar_raw = clinvar_raw.drop(columns = drop_cols)\n",
    "clinvar_raw.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Only interested in cancer related rows right now, filter on cancer related conditions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(422117, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>PhenotypeList</th>\n",
       "      <th>VariationID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>Hemochromatosis type 1|Hereditary cancer-predi...</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>Hemochromatosis type 1|Hereditary cancer-predi...</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>TMEM127</td>\n",
       "      <td>121908830</td>\n",
       "      <td>Pheochromocytoma, susceptibility to|Pheochromo...</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>TMEM127</td>\n",
       "      <td>121908830</td>\n",
       "      <td>Pheochromocytoma, susceptibility to|Pheochromo...</td>\n",
       "      <td>108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>528</th>\n",
       "      <td>KLHDC8B</td>\n",
       "      <td>387906223</td>\n",
       "      <td>Classic Hodgkin lymphoma</td>\n",
       "      <td>273</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    GeneSymbol  RS# (dbSNP)  \\\n",
       "12         HFE      1800562   \n",
       "13         HFE      1800562   \n",
       "206    TMEM127    121908830   \n",
       "207    TMEM127    121908830   \n",
       "528    KLHDC8B    387906223   \n",
       "\n",
       "                                         PhenotypeList  VariationID  \n",
       "12   Hemochromatosis type 1|Hereditary cancer-predi...            9  \n",
       "13   Hemochromatosis type 1|Hereditary cancer-predi...            9  \n",
       "206  Pheochromocytoma, susceptibility to|Pheochromo...          108  \n",
       "207  Pheochromocytoma, susceptibility to|Pheochromo...          108  \n",
       "528                           Classic Hodgkin lymphoma          273  "
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_related = ['cancer', 'carcinoma', 'leukemia', 'tumor', 'malignancy', 'glioblastoma',\n",
    "                'melanoma', 'lymphoma', 'sarcoma']\n",
    "\n",
    "clinvar_raw = clinvar_raw[clinvar_raw['PhenotypeList'].str.contains('|'.join(cancer_related), case = False, na = False)]  \n",
    "print(clinvar_raw.shape)\n",
    "clinvar_raw.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Explode on phenotype list by `|` then `;` and then re-filter (filtering is done again versus doing it before original filter because of sheer size of data)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>PhenotypeList</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PhenotypeList_new</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>Hemochromatosis type 1|Hereditary cancer-predi...</td>\n",
       "      <td>9</td>\n",
       "      <td>[Hemochromatosis type 1, Hereditary cancer-pre...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>Hemochromatosis type 1|Hereditary cancer-predi...</td>\n",
       "      <td>9</td>\n",
       "      <td>[Hemochromatosis type 1, Hereditary cancer-pre...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>TMEM127</td>\n",
       "      <td>121908830</td>\n",
       "      <td>Pheochromocytoma, susceptibility to|Pheochromo...</td>\n",
       "      <td>108</td>\n",
       "      <td>[Pheochromocytoma, susceptibility to, Pheochro...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>TMEM127</td>\n",
       "      <td>121908830</td>\n",
       "      <td>Pheochromocytoma, susceptibility to|Pheochromo...</td>\n",
       "      <td>108</td>\n",
       "      <td>[Pheochromocytoma, susceptibility to, Pheochro...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>528</th>\n",
       "      <td>KLHDC8B</td>\n",
       "      <td>387906223</td>\n",
       "      <td>Classic Hodgkin lymphoma</td>\n",
       "      <td>273</td>\n",
       "      <td>[Classic Hodgkin lymphoma]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    GeneSymbol  RS# (dbSNP)  \\\n",
       "12         HFE      1800562   \n",
       "13         HFE      1800562   \n",
       "206    TMEM127    121908830   \n",
       "207    TMEM127    121908830   \n",
       "528    KLHDC8B    387906223   \n",
       "\n",
       "                                         PhenotypeList  VariationID  \\\n",
       "12   Hemochromatosis type 1|Hereditary cancer-predi...            9   \n",
       "13   Hemochromatosis type 1|Hereditary cancer-predi...            9   \n",
       "206  Pheochromocytoma, susceptibility to|Pheochromo...          108   \n",
       "207  Pheochromocytoma, susceptibility to|Pheochromo...          108   \n",
       "528                           Classic Hodgkin lymphoma          273   \n",
       "\n",
       "                                     PhenotypeList_new  \n",
       "12   [Hemochromatosis type 1, Hereditary cancer-pre...  \n",
       "13   [Hemochromatosis type 1, Hereditary cancer-pre...  \n",
       "206  [Pheochromocytoma, susceptibility to, Pheochro...  \n",
       "207  [Pheochromocytoma, susceptibility to, Pheochro...  \n",
       "528                         [Classic Hodgkin lymphoma]  "
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_raw['PhenotypeList_new'] = clinvar_raw['PhenotypeList'].str.split('|')\n",
    "clinvar_raw.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(822841, 5)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>PhenotypeList</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PhenotypeList_new</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>Hemochromatosis type 1|Hereditary cancer-predi...</td>\n",
       "      <td>9</td>\n",
       "      <td>Hemochromatosis type 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>Hemochromatosis type 1|Hereditary cancer-predi...</td>\n",
       "      <td>9</td>\n",
       "      <td>Hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>Hemochromatosis type 1|Hereditary cancer-predi...</td>\n",
       "      <td>9</td>\n",
       "      <td>not provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>Hemochromatosis type 1|Hereditary cancer-predi...</td>\n",
       "      <td>9</td>\n",
       "      <td>Hereditary hemochromatosis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>Hemochromatosis type 1|Hereditary cancer-predi...</td>\n",
       "      <td>9</td>\n",
       "      <td>Porphyrinuria;Cutaneous photosensitivity</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   GeneSymbol  RS# (dbSNP)                                      PhenotypeList  \\\n",
       "12        HFE      1800562  Hemochromatosis type 1|Hereditary cancer-predi...   \n",
       "12        HFE      1800562  Hemochromatosis type 1|Hereditary cancer-predi...   \n",
       "12        HFE      1800562  Hemochromatosis type 1|Hereditary cancer-predi...   \n",
       "12        HFE      1800562  Hemochromatosis type 1|Hereditary cancer-predi...   \n",
       "12        HFE      1800562  Hemochromatosis type 1|Hereditary cancer-predi...   \n",
       "\n",
       "    VariationID                         PhenotypeList_new  \n",
       "12            9                    Hemochromatosis type 1  \n",
       "12            9   Hereditary cancer-predisposing syndrome  \n",
       "12            9                              not provided  \n",
       "12            9                Hereditary hemochromatosis  \n",
       "12            9  Porphyrinuria;Cutaneous photosensitivity  "
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# explode phenotype list new \n",
    "clinvar_raw = clinvar_raw.explode('PhenotypeList_new')\n",
    "print(clinvar_raw.shape)\n",
    "clinvar_raw.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PhenotypeList_new</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Hemochromatosis type 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>not provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Hereditary hemochromatosis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Porphyrinuria;Cutaneous photosensitivity</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   GeneSymbol  RS# (dbSNP)  VariationID  \\\n",
       "12        HFE      1800562            9   \n",
       "12        HFE      1800562            9   \n",
       "12        HFE      1800562            9   \n",
       "12        HFE      1800562            9   \n",
       "12        HFE      1800562            9   \n",
       "\n",
       "                           PhenotypeList_new  \n",
       "12                    Hemochromatosis type 1  \n",
       "12   Hereditary cancer-predisposing syndrome  \n",
       "12                              not provided  \n",
       "12                Hereditary hemochromatosis  \n",
       "12  Porphyrinuria;Cutaneous photosensitivity  "
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# drop old phenotype list column\n",
    "clinvar_raw = clinvar_raw.drop(columns = ['PhenotypeList'])\n",
    "clinvar_raw.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PhenotypeList_new</th>\n",
       "      <th>PhenotypeList</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Hemochromatosis type 1</td>\n",
       "      <td>[Hemochromatosis type 1]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Hereditary cancer-predisposing syndrome</td>\n",
       "      <td>[Hereditary cancer-predisposing syndrome]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>not provided</td>\n",
       "      <td>[not provided]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Hereditary hemochromatosis</td>\n",
       "      <td>[Hereditary hemochromatosis]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Porphyrinuria;Cutaneous photosensitivity</td>\n",
       "      <td>[Porphyrinuria, Cutaneous photosensitivity]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   GeneSymbol  RS# (dbSNP)  VariationID  \\\n",
       "12        HFE      1800562            9   \n",
       "12        HFE      1800562            9   \n",
       "12        HFE      1800562            9   \n",
       "12        HFE      1800562            9   \n",
       "12        HFE      1800562            9   \n",
       "\n",
       "                           PhenotypeList_new  \\\n",
       "12                    Hemochromatosis type 1   \n",
       "12   Hereditary cancer-predisposing syndrome   \n",
       "12                              not provided   \n",
       "12                Hereditary hemochromatosis   \n",
       "12  Porphyrinuria;Cutaneous photosensitivity   \n",
       "\n",
       "                                  PhenotypeList  \n",
       "12                     [Hemochromatosis type 1]  \n",
       "12    [Hereditary cancer-predisposing syndrome]  \n",
       "12                               [not provided]  \n",
       "12                 [Hereditary hemochromatosis]  \n",
       "12  [Porphyrinuria, Cutaneous photosensitivity]  "
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_raw['PhenotypeList'] = clinvar_raw['PhenotypeList_new'].str.split(';')\n",
    "clinvar_raw.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(882110, 5)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PhenotypeList_new</th>\n",
       "      <th>PhenotypeList</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Hemochromatosis type 1</td>\n",
       "      <td>Hemochromatosis type 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Hereditary cancer-predisposing syndrome</td>\n",
       "      <td>Hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>not provided</td>\n",
       "      <td>not provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Hereditary hemochromatosis</td>\n",
       "      <td>Hereditary hemochromatosis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Porphyrinuria;Cutaneous photosensitivity</td>\n",
       "      <td>Porphyrinuria</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   GeneSymbol  RS# (dbSNP)  VariationID  \\\n",
       "12        HFE      1800562            9   \n",
       "12        HFE      1800562            9   \n",
       "12        HFE      1800562            9   \n",
       "12        HFE      1800562            9   \n",
       "12        HFE      1800562            9   \n",
       "\n",
       "                           PhenotypeList_new  \\\n",
       "12                    Hemochromatosis type 1   \n",
       "12   Hereditary cancer-predisposing syndrome   \n",
       "12                              not provided   \n",
       "12                Hereditary hemochromatosis   \n",
       "12  Porphyrinuria;Cutaneous photosensitivity   \n",
       "\n",
       "                              PhenotypeList  \n",
       "12                   Hemochromatosis type 1  \n",
       "12  Hereditary cancer-predisposing syndrome  \n",
       "12                             not provided  \n",
       "12               Hereditary hemochromatosis  \n",
       "12                            Porphyrinuria  "
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# explode phenotype list again\n",
    "clinvar_raw = clinvar_raw.explode('PhenotypeList')\n",
    "print(clinvar_raw.shape)\n",
    "clinvar_raw.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PhenotypeList</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Hemochromatosis type 1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>not provided</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Hereditary hemochromatosis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Porphyrinuria</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   GeneSymbol  RS# (dbSNP)  VariationID  \\\n",
       "12        HFE      1800562            9   \n",
       "12        HFE      1800562            9   \n",
       "12        HFE      1800562            9   \n",
       "12        HFE      1800562            9   \n",
       "12        HFE      1800562            9   \n",
       "\n",
       "                              PhenotypeList  \n",
       "12                   Hemochromatosis type 1  \n",
       "12  Hereditary cancer-predisposing syndrome  \n",
       "12                             not provided  \n",
       "12               Hereditary hemochromatosis  \n",
       "12                            Porphyrinuria  "
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# drop old phenotype list column\n",
    "clinvar_raw = clinvar_raw.drop(columns = ['PhenotypeList_new'])\n",
    "clinvar_raw.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(563364, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneSymbol</th>\n",
       "      <th>RS# (dbSNP)</th>\n",
       "      <th>VariationID</th>\n",
       "      <th>PhenotypeList</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HFE</td>\n",
       "      <td>1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>TMEM127</td>\n",
       "      <td>121908830</td>\n",
       "      <td>108</td>\n",
       "      <td>Hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>TMEM127</td>\n",
       "      <td>121908830</td>\n",
       "      <td>108</td>\n",
       "      <td>Hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>528</th>\n",
       "      <td>KLHDC8B</td>\n",
       "      <td>387906223</td>\n",
       "      <td>273</td>\n",
       "      <td>Classic Hodgkin lymphoma</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    GeneSymbol  RS# (dbSNP)  VariationID  \\\n",
       "12         HFE      1800562            9   \n",
       "13         HFE      1800562            9   \n",
       "206    TMEM127    121908830          108   \n",
       "207    TMEM127    121908830          108   \n",
       "528    KLHDC8B    387906223          273   \n",
       "\n",
       "                               PhenotypeList  \n",
       "12   Hereditary cancer-predisposing syndrome  \n",
       "13   Hereditary cancer-predisposing syndrome  \n",
       "206  Hereditary cancer-predisposing syndrome  \n",
       "207  Hereditary cancer-predisposing syndrome  \n",
       "528                 Classic Hodgkin lymphoma  "
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# re-filter for cancer related diseases\n",
    "clinvar_raw = clinvar_raw[clinvar_raw['PhenotypeList'].str.contains('|'.join(cancer_related), case = False, na = False)]\n",
    "print(clinvar_raw.shape)\n",
    "clinvar_raw.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GeneSymbol       0\n",
       "RS# (dbSNP)      0\n",
       "VariationID      0\n",
       "PhenotypeList    0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_raw.isnull().sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Re-format rs ID column."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "12           rs1800562\n",
       "13           rs1800562\n",
       "206        rs121908830\n",
       "207        rs121908830\n",
       "528        rs387906223\n",
       "              ...     \n",
       "4807791           rs-1\n",
       "4807792           rs-1\n",
       "4807793           rs-1\n",
       "4807794           rs-1\n",
       "4807795           rs-1\n",
       "Name: RS# (dbSNP), Length: 563364, dtype: object"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_raw['RS# (dbSNP)'] = 'rs' + clinvar_raw['RS# (dbSNP)'].astype(str)\n",
    "clinvar_raw['RS# (dbSNP)']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(451062, 4)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "RS# (dbSNP)\n",
       "rs11540652      132\n",
       "rs28934576      122\n",
       "rs587778720     112\n",
       "rs786201057     104\n",
       "rs587780070     100\n",
       "               ... \n",
       "rs2137450852      1\n",
       "rs552306042       1\n",
       "rs2137457893      1\n",
       "rs2137458677      1\n",
       "rs45445694        1\n",
       "Name: count, Length: 129423, dtype: int64"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_raw = clinvar_raw[~clinvar_raw['RS# (dbSNP)'].str.contains('-1')]\n",
    "print(clinvar_raw.shape)\n",
    "clinvar_raw['RS# (dbSNP)'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Rename columns. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>rs1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HFE</td>\n",
       "      <td>rs1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>Hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>TMEM127</td>\n",
       "      <td>rs121908830</td>\n",
       "      <td>108</td>\n",
       "      <td>Hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>TMEM127</td>\n",
       "      <td>rs121908830</td>\n",
       "      <td>108</td>\n",
       "      <td>Hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>528</th>\n",
       "      <td>KLHDC8B</td>\n",
       "      <td>rs387906223</td>\n",
       "      <td>273</td>\n",
       "      <td>Classic Hodgkin lymphoma</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene        rs_id  variation_id  \\\n",
       "12       HFE    rs1800562             9   \n",
       "13       HFE    rs1800562             9   \n",
       "206  TMEM127  rs121908830           108   \n",
       "207  TMEM127  rs121908830           108   \n",
       "528  KLHDC8B  rs387906223           273   \n",
       "\n",
       "                                     disease  \n",
       "12   Hereditary cancer-predisposing syndrome  \n",
       "13   Hereditary cancer-predisposing syndrome  \n",
       "206  Hereditary cancer-predisposing syndrome  \n",
       "207  Hereditary cancer-predisposing syndrome  \n",
       "528                 Classic Hodgkin lymphoma  "
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols = ['gene', 'rs_id', 'variation_id', 'disease']\n",
    "clinvar_raw.columns = cols\n",
    "clinvar_raw.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Normalize condition values before isolating."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HFE</td>\n",
       "      <td>rs1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HFE</td>\n",
       "      <td>rs1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>TMEM127</td>\n",
       "      <td>rs121908830</td>\n",
       "      <td>108</td>\n",
       "      <td>hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>207</th>\n",
       "      <td>TMEM127</td>\n",
       "      <td>rs121908830</td>\n",
       "      <td>108</td>\n",
       "      <td>hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>528</th>\n",
       "      <td>KLHDC8B</td>\n",
       "      <td>rs387906223</td>\n",
       "      <td>273</td>\n",
       "      <td>classic hodgkin lymphoma</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene        rs_id  variation_id  \\\n",
       "12       HFE    rs1800562             9   \n",
       "13       HFE    rs1800562             9   \n",
       "206  TMEM127  rs121908830           108   \n",
       "207  TMEM127  rs121908830           108   \n",
       "528  KLHDC8B  rs387906223           273   \n",
       "\n",
       "                                     disease  \n",
       "12   hereditary cancer-predisposing syndrome  \n",
       "13   hereditary cancer-predisposing syndrome  \n",
       "206  hereditary cancer-predisposing syndrome  \n",
       "207  hereditary cancer-predisposing syndrome  \n",
       "528                 classic hodgkin lymphoma  "
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_raw['disease'] = clinvar_raw['disease'].str.lower().str.strip()\n",
    "clinvar_raw.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Isolate condition data for cleaning. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "conditions = clinvar_raw['disease']\n",
    "conditions = conditions.drop_duplicates()\n",
    "conditions.to_csv('../home/data/processed_data/GlyGen/clinvar/clinvar_conditions.tsv', sep = '\\t', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(414,)"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conditions.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Save processed data and isolate conditions to clean manually. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_raw.to_csv('../home/data/processed_data/GlyGen/clinvar/clinvar_processed.tsv', sep = '\\t', index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Secondary Wrangling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(451062, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HFE</td>\n",
       "      <td>rs1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HFE</td>\n",
       "      <td>rs1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TMEM127</td>\n",
       "      <td>rs121908830</td>\n",
       "      <td>108</td>\n",
       "      <td>hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TMEM127</td>\n",
       "      <td>rs121908830</td>\n",
       "      <td>108</td>\n",
       "      <td>hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KLHDC8B</td>\n",
       "      <td>rs387906223</td>\n",
       "      <td>273</td>\n",
       "      <td>classic hodgkin lymphoma</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene        rs_id  variation_id                                  disease\n",
       "0      HFE    rs1800562             9  hereditary cancer-predisposing syndrome\n",
       "1      HFE    rs1800562             9  hereditary cancer-predisposing syndrome\n",
       "2  TMEM127  rs121908830           108  hereditary cancer-predisposing syndrome\n",
       "3  TMEM127  rs121908830           108  hereditary cancer-predisposing syndrome\n",
       "4  KLHDC8B  rs387906223           273                 classic hodgkin lymphoma"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_processed = pd.read_csv('../home/data/processed_data/GlyGen/clinvar/clinvar_processed.tsv', sep = '\\t')\n",
    "print(clinvar_processed.shape)\n",
    "clinvar_processed.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "gene            0\n",
       "rs_id           0\n",
       "variation_id    0\n",
       "disease         0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_processed.isnull().sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load cleaned conditions. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_conditions = pd.read_csv('../home/data/processed_data/GlyGen/clinvar/clinvar_conditions.tsv', sep = '\\t')\n",
    "clinvar_drop_conditions = pd.read_csv('../home/data/processed_data/GlyGen/clinvar/clinvar_drop_conditions.tsv', sep = '\\t')\n",
    "clinvar_explode_conditions = pd.read_csv('../home/data/processed_data/GlyGen/clinvar/clinvar_explode_conditions.tsv', sep = '\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Drop conditions that are in the drop list."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [],
   "source": [
    "drop_conditions = set(clinvar_drop_conditions['condition'])\n",
    "clinvar_processed = clinvar_processed[~clinvar_processed['disease'].isin(drop_conditions)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Isolate single and list conditions. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(4228, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>330</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs587776547</td>\n",
       "      <td>3019</td>\n",
       "      <td>breast and/or ovarian cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>335</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs587776547</td>\n",
       "      <td>3019</td>\n",
       "      <td>breast and/or ovarian cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>338</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs774925473</td>\n",
       "      <td>3021</td>\n",
       "      <td>breast and/or ovarian cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs774925473</td>\n",
       "      <td>3021</td>\n",
       "      <td>breast and/or ovarian cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs28904921</td>\n",
       "      <td>3023</td>\n",
       "      <td>breast and/or ovarian cancer</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    gene        rs_id  variation_id                       disease\n",
       "330  ATM  rs587776547          3019  breast and/or ovarian cancer\n",
       "335  ATM  rs587776547          3019  breast and/or ovarian cancer\n",
       "338  ATM  rs774925473          3021  breast and/or ovarian cancer\n",
       "342  ATM  rs774925473          3021  breast and/or ovarian cancer\n",
       "353  ATM   rs28904921          3023  breast and/or ovarian cancer"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_explode_condition_rows = clinvar_processed[clinvar_processed['disease'].isin(set(clinvar_explode_conditions['raw_disease']))]\n",
    "print(clinvar_explode_condition_rows.shape)\n",
    "clinvar_explode_condition_rows.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(445544, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HFE</td>\n",
       "      <td>rs1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HFE</td>\n",
       "      <td>rs1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TMEM127</td>\n",
       "      <td>rs121908830</td>\n",
       "      <td>108</td>\n",
       "      <td>hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TMEM127</td>\n",
       "      <td>rs121908830</td>\n",
       "      <td>108</td>\n",
       "      <td>hereditary cancer-predisposing syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KLHDC8B</td>\n",
       "      <td>rs387906223</td>\n",
       "      <td>273</td>\n",
       "      <td>classic hodgkin lymphoma</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene        rs_id  variation_id                                  disease\n",
       "0      HFE    rs1800562             9  hereditary cancer-predisposing syndrome\n",
       "1      HFE    rs1800562             9  hereditary cancer-predisposing syndrome\n",
       "2  TMEM127  rs121908830           108  hereditary cancer-predisposing syndrome\n",
       "3  TMEM127  rs121908830           108  hereditary cancer-predisposing syndrome\n",
       "4  KLHDC8B  rs387906223           273                 classic hodgkin lymphoma"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_single_condition_rows = clinvar_processed[clinvar_processed['disease'].isin(set(clinvar_conditions['disease']))]\n",
    "print(clinvar_single_condition_rows.shape)\n",
    "clinvar_single_condition_rows.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sanity check\n",
    "assert(clinvar_processed.shape[0] == clinvar_explode_condition_rows.shape[0] + clinvar_single_condition_rows.shape[0])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Map single condition rows."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>condition</th>\n",
       "      <th>doid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>353797</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>rs730881472</td>\n",
       "      <td>950867</td>\n",
       "      <td>hereditary cancer-predisposing syndrome</td>\n",
       "      <td>cancer</td>\n",
       "      <td>:162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69981</th>\n",
       "      <td>PALB2</td>\n",
       "      <td>rs730881867</td>\n",
       "      <td>182740</td>\n",
       "      <td>familial cancer of breast</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>:1612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>287787</th>\n",
       "      <td>ALK</td>\n",
       "      <td>rs552187705</td>\n",
       "      <td>702961</td>\n",
       "      <td>hereditary cancer-predisposing syndrome</td>\n",
       "      <td>cancer</td>\n",
       "      <td>:162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304607</th>\n",
       "      <td>MET</td>\n",
       "      <td>rs1584914144</td>\n",
       "      <td>818832</td>\n",
       "      <td>renal cell carcinoma</td>\n",
       "      <td>renal cell carcinoma</td>\n",
       "      <td>:4450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92176</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>rs863224762</td>\n",
       "      <td>216670</td>\n",
       "      <td>hereditary breast ovarian cancer syndrome</td>\n",
       "      <td>hereditary breast ovarian cancer syndrome</td>\n",
       "      <td>:5683</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene         rs_id  variation_id  \\\n",
       "353797  BRCA1   rs730881472        950867   \n",
       "69981   PALB2   rs730881867        182740   \n",
       "287787    ALK   rs552187705        702961   \n",
       "304607    MET  rs1584914144        818832   \n",
       "92176   BRCA1   rs863224762        216670   \n",
       "\n",
       "                                          disease  \\\n",
       "353797    hereditary cancer-predisposing syndrome   \n",
       "69981                   familial cancer of breast   \n",
       "287787    hereditary cancer-predisposing syndrome   \n",
       "304607                       renal cell carcinoma   \n",
       "92176   hereditary breast ovarian cancer syndrome   \n",
       "\n",
       "                                        condition   doid  \n",
       "353797                                     cancer   :162  \n",
       "69981                               breast cancer  :1612  \n",
       "287787                                     cancer   :162  \n",
       "304607                       renal cell carcinoma  :4450  \n",
       "92176   hereditary breast ovarian cancer syndrome  :5683  "
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_single_condition_rows = clinvar_single_condition_rows.merge(clinvar_conditions, how='left', on='disease')\n",
    "clinvar_single_condition_rows.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "gene                0\n",
       "rs_id               0\n",
       "variation_id        0\n",
       "disease             0\n",
       "condition       35923\n",
       "doid                0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_single_condition_rows.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['gene', 'rs_id', 'variation_id', 'disease', 'condition', 'doid'], dtype='object')"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_single_condition_rows.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>condition</th>\n",
       "      <th>doid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>399297</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>rs2154475121</td>\n",
       "      <td>1171931</td>\n",
       "      <td>hereditary cancer-predisposing syndrome</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188671</th>\n",
       "      <td>RET</td>\n",
       "      <td>rs191769748</td>\n",
       "      <td>477367</td>\n",
       "      <td>familial medullary thyroid carcinoma</td>\n",
       "      <td>familial medullary thyroid carcinoma</td>\n",
       "      <td>DOID:0050547</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77024</th>\n",
       "      <td>MRE11</td>\n",
       "      <td>rs786202951</td>\n",
       "      <td>186437</td>\n",
       "      <td>hereditary cancer-predisposing syndrome</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>285187</th>\n",
       "      <td>HOXB13</td>\n",
       "      <td>rs1597934943</td>\n",
       "      <td>690580</td>\n",
       "      <td>prostate cancer, hereditary, 1</td>\n",
       "      <td>prostate cancer</td>\n",
       "      <td>DOID:10283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26960</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>rs397508931</td>\n",
       "      <td>54441</td>\n",
       "      <td>breast-ovarian cancer, familial, susceptibilit...</td>\n",
       "      <td>hereditary breast ovarian cancer syndrome</td>\n",
       "      <td>DOID:5683</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          gene         rs_id  variation_id  \\\n",
       "399297   BRCA1  rs2154475121       1171931   \n",
       "188671     RET   rs191769748        477367   \n",
       "77024    MRE11   rs786202951        186437   \n",
       "285187  HOXB13  rs1597934943        690580   \n",
       "26960    BRCA1   rs397508931         54441   \n",
       "\n",
       "                                                  disease  \\\n",
       "399297            hereditary cancer-predisposing syndrome   \n",
       "188671               familial medullary thyroid carcinoma   \n",
       "77024             hereditary cancer-predisposing syndrome   \n",
       "285187                     prostate cancer, hereditary, 1   \n",
       "26960   breast-ovarian cancer, familial, susceptibilit...   \n",
       "\n",
       "                                        condition          doid  \n",
       "399297                                     cancer      DOID:162  \n",
       "188671       familial medullary thyroid carcinoma  DOID:0050547  \n",
       "77024                                      cancer      DOID:162  \n",
       "285187                            prostate cancer    DOID:10283  \n",
       "26960   hereditary breast ovarian cancer syndrome     DOID:5683  "
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_single_condition_rows['doid'] = 'DOID' + clinvar_single_condition_rows['doid'].astype(str)\n",
    "clinvar_single_condition_rows.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>condition</th>\n",
       "      <th>doid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>123080</th>\n",
       "      <td>SMARCA4</td>\n",
       "      <td>rs146141457</td>\n",
       "      <td>238531</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DOID:</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>390437</th>\n",
       "      <td>SMARCA4</td>\n",
       "      <td>rs765751495</td>\n",
       "      <td>1108552</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>393230</th>\n",
       "      <td>DICER1</td>\n",
       "      <td>rs552975520</td>\n",
       "      <td>1139621</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DOID:</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>329659</th>\n",
       "      <td>TSC2</td>\n",
       "      <td>rs397515241</td>\n",
       "      <td>838204</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369137</th>\n",
       "      <td>ALK</td>\n",
       "      <td>rs753812499</td>\n",
       "      <td>1023840</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           gene        rs_id  variation_id condition      doid\n",
       "123080  SMARCA4  rs146141457        238531       NaN     DOID:\n",
       "390437  SMARCA4  rs765751495       1108552    cancer  DOID:162\n",
       "393230   DICER1  rs552975520       1139621       NaN     DOID:\n",
       "329659     TSC2  rs397515241        838204    cancer  DOID:162\n",
       "369137      ALK  rs753812499       1023840    cancer  DOID:162"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_single_condition_rows = clinvar_single_condition_rows.drop('disease', axis=1)\n",
    "clinvar_single_condition_rows.sample(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Map multiple condition rows."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>292330</th>\n",
       "      <td>ATR</td>\n",
       "      <td>rs1397771631</td>\n",
       "      <td>702854</td>\n",
       "      <td>familial cutaneous telangiectasia and orophary...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51408</th>\n",
       "      <td>PMS2</td>\n",
       "      <td>rs374704824</td>\n",
       "      <td>127793</td>\n",
       "      <td>breast and/or ovarian cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35049</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>rs80357066</td>\n",
       "      <td>55769</td>\n",
       "      <td>breast and/or ovarian cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>373973</th>\n",
       "      <td>MSH2</td>\n",
       "      <td>rs1203462814</td>\n",
       "      <td>1021808</td>\n",
       "      <td>breast and/or ovarian cancer</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57531</th>\n",
       "      <td>CDH1</td>\n",
       "      <td>rs115817750</td>\n",
       "      <td>136067</td>\n",
       "      <td>breast and/or ovarian cancer</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         gene         rs_id  variation_id  \\\n",
       "292330    ATR  rs1397771631        702854   \n",
       "51408    PMS2   rs374704824        127793   \n",
       "35049   BRCA1    rs80357066         55769   \n",
       "373973   MSH2  rs1203462814       1021808   \n",
       "57531    CDH1   rs115817750        136067   \n",
       "\n",
       "                                                  disease  \n",
       "292330  familial cutaneous telangiectasia and orophary...  \n",
       "51408                        breast and/or ovarian cancer  \n",
       "35049                        breast and/or ovarian cancer  \n",
       "373973                       breast and/or ovarian cancer  \n",
       "57531                        breast and/or ovarian cancer  "
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_explode_condition_rows.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>raw_disease</th>\n",
       "      <th>list_diseases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>colorectal / endometrial cancer</td>\n",
       "      <td>colorectal cancer:9256,endometrial cancer:1380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>increased risk to develop myelodysplastic synd...</td>\n",
       "      <td>myelodysplastic syndrome:0050908,acute myeloid...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>tumor predisposition syndrome 4, breast/prosta...</td>\n",
       "      <td>breast cancer:1612,prostate cancer:10283,color...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>breast and/or ovarian cancer</td>\n",
       "      <td>breast cancer:1612,ovarian cancer:2394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>prostate cancer/brain cancer susceptibility</td>\n",
       "      <td>prostate cancer:10283,brain cancer:1319</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         raw_disease  \\\n",
       "4                    colorectal / endometrial cancer   \n",
       "8  increased risk to develop myelodysplastic synd...   \n",
       "2  tumor predisposition syndrome 4, breast/prosta...   \n",
       "0                       breast and/or ovarian cancer   \n",
       "3        prostate cancer/brain cancer susceptibility   \n",
       "\n",
       "                                       list_diseases  \n",
       "4     colorectal cancer:9256,endometrial cancer:1380  \n",
       "8  myelodysplastic syndrome:0050908,acute myeloid...  \n",
       "2  breast cancer:1612,prostate cancer:10283,color...  \n",
       "0             breast cancer:1612,ovarian cancer:2394  \n",
       "3            prostate cancer:10283,brain cancer:1319  "
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_explode_conditions.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>raw_disease</th>\n",
       "      <th>list_diseases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>844</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>rs397507907</td>\n",
       "      <td>52298</td>\n",
       "      <td>breast and/or ovarian cancer</td>\n",
       "      <td>breast and/or ovarian cancer</td>\n",
       "      <td>breast cancer:1612,ovarian cancer:2394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1992</th>\n",
       "      <td>RAD51D</td>\n",
       "      <td>rs587781756</td>\n",
       "      <td>141452</td>\n",
       "      <td>breast and/or ovarian cancer</td>\n",
       "      <td>breast and/or ovarian cancer</td>\n",
       "      <td>breast cancer:1612,ovarian cancer:2394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3872</th>\n",
       "      <td>ATR</td>\n",
       "      <td>rs778499519</td>\n",
       "      <td>1127830</td>\n",
       "      <td>familial cutaneous telangiectasia and orophary...</td>\n",
       "      <td>familial cutaneous telangiectasia and orophary...</td>\n",
       "      <td>[TODO]:[TODO],[TODO]:[TODO]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1138</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>rs80357066</td>\n",
       "      <td>55769</td>\n",
       "      <td>breast and/or ovarian cancer</td>\n",
       "      <td>breast and/or ovarian cancer</td>\n",
       "      <td>breast cancer:1612,ovarian cancer:2394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3022</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs767070325</td>\n",
       "      <td>378989</td>\n",
       "      <td>breast and/or ovarian cancer</td>\n",
       "      <td>breast and/or ovarian cancer</td>\n",
       "      <td>breast cancer:1612,ovarian cancer:2394</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        gene        rs_id  variation_id  \\\n",
       "844    BRCA2  rs397507907         52298   \n",
       "1992  RAD51D  rs587781756        141452   \n",
       "3872     ATR  rs778499519       1127830   \n",
       "1138   BRCA1   rs80357066         55769   \n",
       "3022     ATM  rs767070325        378989   \n",
       "\n",
       "                                                disease  \\\n",
       "844                        breast and/or ovarian cancer   \n",
       "1992                       breast and/or ovarian cancer   \n",
       "3872  familial cutaneous telangiectasia and orophary...   \n",
       "1138                       breast and/or ovarian cancer   \n",
       "3022                       breast and/or ovarian cancer   \n",
       "\n",
       "                                            raw_disease  \\\n",
       "844                        breast and/or ovarian cancer   \n",
       "1992                       breast and/or ovarian cancer   \n",
       "3872  familial cutaneous telangiectasia and orophary...   \n",
       "1138                       breast and/or ovarian cancer   \n",
       "3022                       breast and/or ovarian cancer   \n",
       "\n",
       "                               list_diseases  \n",
       "844   breast cancer:1612,ovarian cancer:2394  \n",
       "1992  breast cancer:1612,ovarian cancer:2394  \n",
       "3872             [TODO]:[TODO],[TODO]:[TODO]  \n",
       "1138  breast cancer:1612,ovarian cancer:2394  \n",
       "3022  breast cancer:1612,ovarian cancer:2394  "
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_explode_condition_rows = clinvar_explode_condition_rows.merge(clinvar_explode_conditions, how='left', left_on='disease', right_on='raw_disease')\n",
    "clinvar_explode_condition_rows.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "gene             0\n",
       "rs_id            0\n",
       "variation_id     0\n",
       "disease          0\n",
       "raw_disease      0\n",
       "list_diseases    0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_explode_condition_rows.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>list_diseases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs587776547</td>\n",
       "      <td>3019</td>\n",
       "      <td>breast cancer:1612,ovarian cancer:2394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs587776547</td>\n",
       "      <td>3019</td>\n",
       "      <td>breast cancer:1612,ovarian cancer:2394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs774925473</td>\n",
       "      <td>3021</td>\n",
       "      <td>breast cancer:1612,ovarian cancer:2394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs774925473</td>\n",
       "      <td>3021</td>\n",
       "      <td>breast cancer:1612,ovarian cancer:2394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs28904921</td>\n",
       "      <td>3023</td>\n",
       "      <td>breast cancer:1612,ovarian cancer:2394</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene        rs_id  variation_id                           list_diseases\n",
       "0  ATM  rs587776547          3019  breast cancer:1612,ovarian cancer:2394\n",
       "1  ATM  rs587776547          3019  breast cancer:1612,ovarian cancer:2394\n",
       "2  ATM  rs774925473          3021  breast cancer:1612,ovarian cancer:2394\n",
       "3  ATM  rs774925473          3021  breast cancer:1612,ovarian cancer:2394\n",
       "4  ATM   rs28904921          3023  breast cancer:1612,ovarian cancer:2394"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_explode_condition_rows = clinvar_explode_condition_rows.drop(['disease', 'raw_disease'], axis = 1)\n",
    "clinvar_explode_condition_rows.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>list_diseases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs587776547</td>\n",
       "      <td>3019</td>\n",
       "      <td>[breast cancer:1612, ovarian cancer:2394]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs587776547</td>\n",
       "      <td>3019</td>\n",
       "      <td>[breast cancer:1612, ovarian cancer:2394]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs774925473</td>\n",
       "      <td>3021</td>\n",
       "      <td>[breast cancer:1612, ovarian cancer:2394]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs774925473</td>\n",
       "      <td>3021</td>\n",
       "      <td>[breast cancer:1612, ovarian cancer:2394]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs28904921</td>\n",
       "      <td>3023</td>\n",
       "      <td>[breast cancer:1612, ovarian cancer:2394]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene        rs_id  variation_id                              list_diseases\n",
       "0  ATM  rs587776547          3019  [breast cancer:1612, ovarian cancer:2394]\n",
       "1  ATM  rs587776547          3019  [breast cancer:1612, ovarian cancer:2394]\n",
       "2  ATM  rs774925473          3021  [breast cancer:1612, ovarian cancer:2394]\n",
       "3  ATM  rs774925473          3021  [breast cancer:1612, ovarian cancer:2394]\n",
       "4  ATM   rs28904921          3023  [breast cancer:1612, ovarian cancer:2394]"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_explode_condition_rows['list_diseases'] = clinvar_explode_condition_rows['list_diseases'].str.split(',')\n",
    "clinvar_explode_condition_rows.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(8462, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>list_diseases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs587776547</td>\n",
       "      <td>3019</td>\n",
       "      <td>breast cancer:1612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs587776547</td>\n",
       "      <td>3019</td>\n",
       "      <td>ovarian cancer:2394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs587776547</td>\n",
       "      <td>3019</td>\n",
       "      <td>breast cancer:1612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs587776547</td>\n",
       "      <td>3019</td>\n",
       "      <td>ovarian cancer:2394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs774925473</td>\n",
       "      <td>3021</td>\n",
       "      <td>breast cancer:1612</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene        rs_id  variation_id        list_diseases\n",
       "0  ATM  rs587776547          3019   breast cancer:1612\n",
       "0  ATM  rs587776547          3019  ovarian cancer:2394\n",
       "1  ATM  rs587776547          3019   breast cancer:1612\n",
       "1  ATM  rs587776547          3019  ovarian cancer:2394\n",
       "2  ATM  rs774925473          3021   breast cancer:1612"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_explode_condition_rows = clinvar_explode_condition_rows.explode('list_diseases')\n",
    "print(clinvar_explode_condition_rows.shape)\n",
    "clinvar_explode_condition_rows.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>condition</th>\n",
       "      <th>doid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs587776547</td>\n",
       "      <td>3019</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>DOID:1612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs587776547</td>\n",
       "      <td>3019</td>\n",
       "      <td>ovarian cancer</td>\n",
       "      <td>DOID:2394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs587776547</td>\n",
       "      <td>3019</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>DOID:1612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs587776547</td>\n",
       "      <td>3019</td>\n",
       "      <td>ovarian cancer</td>\n",
       "      <td>DOID:2394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs774925473</td>\n",
       "      <td>3021</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>DOID:1612</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  gene        rs_id  variation_id       condition       doid\n",
       "0  ATM  rs587776547          3019   breast cancer  DOID:1612\n",
       "0  ATM  rs587776547          3019  ovarian cancer  DOID:2394\n",
       "1  ATM  rs587776547          3019   breast cancer  DOID:1612\n",
       "1  ATM  rs587776547          3019  ovarian cancer  DOID:2394\n",
       "2  ATM  rs774925473          3021   breast cancer  DOID:1612"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_explode_condition_rows[['condition', 'doid']] = clinvar_explode_condition_rows['list_diseases'].str.split(':', expand=True, n=1)\n",
    "clinvar_explode_condition_rows['doid'] = 'DOID:' + clinvar_explode_condition_rows['doid'].astype(str)\n",
    "clinvar_explode_condition_rows.drop(columns=['list_diseases'], inplace=True)\n",
    "clinvar_explode_condition_rows.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Recombine data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>condition</th>\n",
       "      <th>doid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>421640</th>\n",
       "      <td>MET</td>\n",
       "      <td>rs2117029370</td>\n",
       "      <td>1430735</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163028</th>\n",
       "      <td>PALB2</td>\n",
       "      <td>rs1060502737</td>\n",
       "      <td>410114</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>DOID:1612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>258727</th>\n",
       "      <td>RAD51D</td>\n",
       "      <td>rs28363283</td>\n",
       "      <td>581983</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135158</th>\n",
       "      <td>MTAP</td>\n",
       "      <td>rs4977735</td>\n",
       "      <td>366307</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DOID:</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259378</th>\n",
       "      <td>GPC3</td>\n",
       "      <td>rs761660909</td>\n",
       "      <td>578171</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DOID:</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48696</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs587779861</td>\n",
       "      <td>127437</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>DOID:1612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280192</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>rs1597864637</td>\n",
       "      <td>639392</td>\n",
       "      <td>hereditary breast ovarian cancer syndrome</td>\n",
       "      <td>DOID:5683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74835</th>\n",
       "      <td>PMS2</td>\n",
       "      <td>rs759192470</td>\n",
       "      <td>185217</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>154308</th>\n",
       "      <td>ABRAXAS1</td>\n",
       "      <td>rs370520589</td>\n",
       "      <td>416705</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103092</th>\n",
       "      <td>MLH1</td>\n",
       "      <td>rs63750016</td>\n",
       "      <td>233750</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            gene         rs_id  variation_id  \\\n",
       "421640       MET  rs2117029370       1430735   \n",
       "163028     PALB2  rs1060502737        410114   \n",
       "258727    RAD51D    rs28363283        581983   \n",
       "135158      MTAP     rs4977735        366307   \n",
       "259378      GPC3   rs761660909        578171   \n",
       "48696        ATM   rs587779861        127437   \n",
       "280192     BRCA1  rs1597864637        639392   \n",
       "74835       PMS2   rs759192470        185217   \n",
       "154308  ABRAXAS1   rs370520589        416705   \n",
       "103092      MLH1    rs63750016        233750   \n",
       "\n",
       "                                        condition       doid  \n",
       "421640                                     cancer   DOID:162  \n",
       "163028                              breast cancer  DOID:1612  \n",
       "258727                                     cancer   DOID:162  \n",
       "135158                                        NaN      DOID:  \n",
       "259378                                        NaN      DOID:  \n",
       "48696                               breast cancer  DOID:1612  \n",
       "280192  hereditary breast ovarian cancer syndrome  DOID:5683  \n",
       "74835                                      cancer   DOID:162  \n",
       "154308                                     cancer   DOID:162  \n",
       "103092                                     cancer   DOID:162  "
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_processed = pd.concat([clinvar_single_condition_rows, clinvar_single_condition_rows])\n",
    "clinvar_processed.sample(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_processed.to_csv('../home/data/processed_data/GlyGen/clinvar/clinvar_final_processed.tsv', sep='\\t', index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Final Mappings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>variation_id</th>\n",
       "      <th>condition</th>\n",
       "      <th>doid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HFE</td>\n",
       "      <td>rs1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HFE</td>\n",
       "      <td>rs1800562</td>\n",
       "      <td>9</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TMEM127</td>\n",
       "      <td>rs121908830</td>\n",
       "      <td>108</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TMEM127</td>\n",
       "      <td>rs121908830</td>\n",
       "      <td>108</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KLHDC8B</td>\n",
       "      <td>rs387906223</td>\n",
       "      <td>273</td>\n",
       "      <td>Hodgkin's lymphoma</td>\n",
       "      <td>DOID:8567</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      gene        rs_id  variation_id           condition       doid\n",
       "0      HFE    rs1800562             9              cancer   DOID:162\n",
       "1      HFE    rs1800562             9              cancer   DOID:162\n",
       "2  TMEM127  rs121908830           108              cancer   DOID:162\n",
       "3  TMEM127  rs121908830           108              cancer   DOID:162\n",
       "4  KLHDC8B  rs387906223           273  Hodgkin's lymphoma  DOID:8567"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_processed = pd.read_csv('../home/data/processed_data/GlyGen/clinvar/clinvar_final_processed.tsv', sep='\\t')\n",
    "clinvar_processed.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biomarker_id</th>\n",
       "      <th>biomarker</th>\n",
       "      <th>assessed_biomarker_entity</th>\n",
       "      <th>assessed_biomarker_entity_id</th>\n",
       "      <th>assessed_entity_type</th>\n",
       "      <th>condition</th>\n",
       "      <th>condition_id</th>\n",
       "      <th>exposure_agent</th>\n",
       "      <th>exposure_agent_id</th>\n",
       "      <th>best_biomarker_role</th>\n",
       "      <th>specimen</th>\n",
       "      <th>specimen_id</th>\n",
       "      <th>loinc_code</th>\n",
       "      <th>evidence_source</th>\n",
       "      <th>evidence</th>\n",
       "      <th>tag</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [biomarker_id, biomarker, assessed_biomarker_entity, assessed_biomarker_entity_id, assessed_entity_type, condition, condition_id, exposure_agent, exposure_agent_id, best_biomarker_role, specimen, specimen_id, loinc_code, evidence_source, evidence, tag]\n",
       "Index: []"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(columns = tcf.TSV_HEADERS)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(891088, 16)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biomarker_id</th>\n",
       "      <th>biomarker</th>\n",
       "      <th>assessed_biomarker_entity</th>\n",
       "      <th>assessed_biomarker_entity_id</th>\n",
       "      <th>assessed_entity_type</th>\n",
       "      <th>condition</th>\n",
       "      <th>condition_id</th>\n",
       "      <th>exposure_agent</th>\n",
       "      <th>exposure_agent_id</th>\n",
       "      <th>best_biomarker_role</th>\n",
       "      <th>specimen</th>\n",
       "      <th>specimen_id</th>\n",
       "      <th>loinc_code</th>\n",
       "      <th>evidence_source</th>\n",
       "      <th>evidence</th>\n",
       "      <th>tag</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>presence of rs1800562 mutation in HFE</td>\n",
       "      <td>HFE</td>\n",
       "      <td>dbSNP:rs1800562</td>\n",
       "      <td>gene</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CLINVAR:9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>presence of rs1800562 mutation in HFE</td>\n",
       "      <td>HFE</td>\n",
       "      <td>dbSNP:rs1800562</td>\n",
       "      <td>gene</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CLINVAR:9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>presence of rs121908830 mutation in TMEM127</td>\n",
       "      <td>TMEM127</td>\n",
       "      <td>dbSNP:rs121908830</td>\n",
       "      <td>gene</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CLINVAR:108</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>presence of rs121908830 mutation in TMEM127</td>\n",
       "      <td>TMEM127</td>\n",
       "      <td>dbSNP:rs121908830</td>\n",
       "      <td>gene</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CLINVAR:108</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>presence of rs387906223 mutation in KLHDC8B</td>\n",
       "      <td>KLHDC8B</td>\n",
       "      <td>dbSNP:rs387906223</td>\n",
       "      <td>gene</td>\n",
       "      <td>Hodgkin's lymphoma</td>\n",
       "      <td>DOID:8567</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CLINVAR:273</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  biomarker_id                                    biomarker  \\\n",
       "0          NaN        presence of rs1800562 mutation in HFE   \n",
       "1          NaN        presence of rs1800562 mutation in HFE   \n",
       "2          NaN  presence of rs121908830 mutation in TMEM127   \n",
       "3          NaN  presence of rs121908830 mutation in TMEM127   \n",
       "4          NaN  presence of rs387906223 mutation in KLHDC8B   \n",
       "\n",
       "  assessed_biomarker_entity assessed_biomarker_entity_id assessed_entity_type  \\\n",
       "0                       HFE              dbSNP:rs1800562                 gene   \n",
       "1                       HFE              dbSNP:rs1800562                 gene   \n",
       "2                   TMEM127            dbSNP:rs121908830                 gene   \n",
       "3                   TMEM127            dbSNP:rs121908830                 gene   \n",
       "4                   KLHDC8B            dbSNP:rs387906223                 gene   \n",
       "\n",
       "            condition condition_id exposure_agent exposure_agent_id  \\\n",
       "0              cancer     DOID:162            NaN               NaN   \n",
       "1              cancer     DOID:162            NaN               NaN   \n",
       "2              cancer     DOID:162            NaN               NaN   \n",
       "3              cancer     DOID:162            NaN               NaN   \n",
       "4  Hodgkin's lymphoma    DOID:8567            NaN               NaN   \n",
       "\n",
       "  best_biomarker_role specimen specimen_id loinc_code evidence_source  \\\n",
       "0                risk      NaN         NaN        NaN       CLINVAR:9   \n",
       "1                risk      NaN         NaN        NaN       CLINVAR:9   \n",
       "2                risk      NaN         NaN        NaN     CLINVAR:108   \n",
       "3                risk      NaN         NaN        NaN     CLINVAR:108   \n",
       "4                risk      NaN         NaN        NaN     CLINVAR:273   \n",
       "\n",
       "  evidence                                                tag  \n",
       "0      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "1      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "2      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "3      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "4      NaN  biomarker;assessed_biomarker_entity_id;assesse...  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['assessed_biomarker_entity'] = clinvar_processed['gene']\n",
    "df['biomarker'] = 'presence of ' + clinvar_processed['rs_id'].astype(str) + ' mutation in ' + df['assessed_biomarker_entity']\n",
    "df['assessed_biomarker_entity_id'] = 'dbSNP:' + clinvar_processed['rs_id'].astype(str)\n",
    "df['assessed_entity_type'] = 'gene'\n",
    "df['condition'] = clinvar_processed['condition'].str.strip()\n",
    "df['condition_id'] = clinvar_processed['doid'].astype(str).str.strip()\n",
    "df['best_biomarker_role'] = 'risk'\n",
    "df['evidence_source'] = 'CLINVAR:' + clinvar_processed['variation_id'].astype(str).str.strip()\n",
    "df['tag'] = 'biomarker;assessed_biomarker_entity_id;assessed_biomarker_entity;condition'\n",
    "\n",
    "print(df.shape)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(211968, 16)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# drop rows that are full duplicates\n",
    "df = df.drop_duplicates()\n",
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biomarker_id</th>\n",
       "      <th>biomarker</th>\n",
       "      <th>assessed_biomarker_entity</th>\n",
       "      <th>assessed_biomarker_entity_id</th>\n",
       "      <th>assessed_entity_type</th>\n",
       "      <th>condition</th>\n",
       "      <th>condition_id</th>\n",
       "      <th>exposure_agent</th>\n",
       "      <th>exposure_agent_id</th>\n",
       "      <th>best_biomarker_role</th>\n",
       "      <th>specimen</th>\n",
       "      <th>specimen_id</th>\n",
       "      <th>loinc_code</th>\n",
       "      <th>evidence_source</th>\n",
       "      <th>evidence</th>\n",
       "      <th>tag</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>64850.0</td>\n",
       "      <td>presence of rs1800562 mutation in HFE</td>\n",
       "      <td>HFE</td>\n",
       "      <td>dbSNP:rs1800562</td>\n",
       "      <td>gene</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CLINVAR:9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>14099.0</td>\n",
       "      <td>presence of rs121908830 mutation in TMEM127</td>\n",
       "      <td>TMEM127</td>\n",
       "      <td>dbSNP:rs121908830</td>\n",
       "      <td>gene</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CLINVAR:108</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>96427.0</td>\n",
       "      <td>presence of rs387906223 mutation in KLHDC8B</td>\n",
       "      <td>KLHDC8B</td>\n",
       "      <td>dbSNP:rs387906223</td>\n",
       "      <td>gene</td>\n",
       "      <td>Hodgkin's lymphoma</td>\n",
       "      <td>DOID:8567</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CLINVAR:273</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>22218.0</td>\n",
       "      <td>presence of rs137854550 mutation in NF1</td>\n",
       "      <td>NF1</td>\n",
       "      <td>dbSNP:rs137854550</td>\n",
       "      <td>gene</td>\n",
       "      <td>juvenile myelomonocytic leukemia</td>\n",
       "      <td>DOID:0050458</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CLINVAR:336</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>22217.0</td>\n",
       "      <td>presence of rs137854550 mutation in NF1</td>\n",
       "      <td>NF1</td>\n",
       "      <td>dbSNP:rs137854550</td>\n",
       "      <td>gene</td>\n",
       "      <td>cancer</td>\n",
       "      <td>DOID:162</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CLINVAR:336</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   biomarker_id                                    biomarker  \\\n",
       "0       64850.0        presence of rs1800562 mutation in HFE   \n",
       "2       14099.0  presence of rs121908830 mutation in TMEM127   \n",
       "4       96427.0  presence of rs387906223 mutation in KLHDC8B   \n",
       "6       22218.0      presence of rs137854550 mutation in NF1   \n",
       "7       22217.0      presence of rs137854550 mutation in NF1   \n",
       "\n",
       "  assessed_biomarker_entity assessed_biomarker_entity_id assessed_entity_type  \\\n",
       "0                       HFE              dbSNP:rs1800562                 gene   \n",
       "2                   TMEM127            dbSNP:rs121908830                 gene   \n",
       "4                   KLHDC8B            dbSNP:rs387906223                 gene   \n",
       "6                       NF1            dbSNP:rs137854550                 gene   \n",
       "7                       NF1            dbSNP:rs137854550                 gene   \n",
       "\n",
       "                          condition  condition_id exposure_agent  \\\n",
       "0                            cancer      DOID:162            NaN   \n",
       "2                            cancer      DOID:162            NaN   \n",
       "4                Hodgkin's lymphoma     DOID:8567            NaN   \n",
       "6  juvenile myelomonocytic leukemia  DOID:0050458            NaN   \n",
       "7                            cancer      DOID:162            NaN   \n",
       "\n",
       "  exposure_agent_id best_biomarker_role specimen specimen_id loinc_code  \\\n",
       "0               NaN                risk      NaN         NaN        NaN   \n",
       "2               NaN                risk      NaN         NaN        NaN   \n",
       "4               NaN                risk      NaN         NaN        NaN   \n",
       "6               NaN                risk      NaN         NaN        NaN   \n",
       "7               NaN                risk      NaN         NaN        NaN   \n",
       "\n",
       "  evidence_source evidence                                                tag  \n",
       "0       CLINVAR:9      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "2     CLINVAR:108      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "4     CLINVAR:273      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "6     CLINVAR:336      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "7     CLINVAR:336      NaN  biomarker;assessed_biomarker_entity_id;assesse...  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['biomarker_id'] = df.groupby(['biomarker', 'assessed_biomarker_entity_id', 'assessed_biomarker_entity', 'condition']).ngroup()\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('../home/data/cleaned_data/GlyGen/clinvar.tsv', sep = '\\t', index = False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
