{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# imports \n",
    "import pandas as pd\n",
    "import numpy as np \n",
    "from data_extraction import DataExtractionHelper\n",
    "\n",
    "# load helper \n",
    "helper = DataExtractionHelper()\n",
    "# remove max column for display output \n",
    "pd.set_option('display.max_columns', None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PUBMEDID</th>\n",
       "      <th>JOURNAL</th>\n",
       "      <th>LINK</th>\n",
       "      <th>STUDY</th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>REPORTED GENE(S)</th>\n",
       "      <th>MAPPED_GENE</th>\n",
       "      <th>UPSTREAM_GENE_ID</th>\n",
       "      <th>DOWNSTREAM_GENE_ID</th>\n",
       "      <th>SNP_GENE_IDS</th>\n",
       "      <th>STRONGEST SNP-RISK ALLELE</th>\n",
       "      <th>SNPS</th>\n",
       "      <th>SNP_ID_CURRENT</th>\n",
       "      <th>CONTEXT</th>\n",
       "      <th>CNV</th>\n",
       "      <th>MAPPED_TRAIT</th>\n",
       "      <th>MAPPED_TRAIT_URI</th>\n",
       "      <th>STUDY ACCESSION</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>26151821</td>\n",
       "      <td>Nat Commun</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/26151821</td>\n",
       "      <td>Genome-wide association study of colorectal ca...</td>\n",
       "      <td>Colorectal cancer</td>\n",
       "      <td>SLC25A26, LRIG1</td>\n",
       "      <td>LRIG1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000144749</td>\n",
       "      <td>rs812481-G</td>\n",
       "      <td>rs812481</td>\n",
       "      <td>812481</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0005842</td>\n",
       "      <td>GCST003017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>26151821</td>\n",
       "      <td>Nat Commun</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/26151821</td>\n",
       "      <td>Genome-wide association study of colorectal ca...</td>\n",
       "      <td>Colorectal cancer</td>\n",
       "      <td>NOS1</td>\n",
       "      <td>NOS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000089250</td>\n",
       "      <td>rs73208120-G</td>\n",
       "      <td>rs73208120</td>\n",
       "      <td>73208120</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0005842</td>\n",
       "      <td>GCST003017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>26151821</td>\n",
       "      <td>Nat Commun</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/26151821</td>\n",
       "      <td>Genome-wide association study of colorectal ca...</td>\n",
       "      <td>Colorectal cancer</td>\n",
       "      <td>ENTPD7, COX15, CUTC, ABCC2, SLC25A28</td>\n",
       "      <td>NKX2-3 - SLC25A28</td>\n",
       "      <td>ENSG00000119919</td>\n",
       "      <td>ENSG00000155287</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs11190164-G</td>\n",
       "      <td>rs11190164</td>\n",
       "      <td>11190164</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0005842</td>\n",
       "      <td>GCST003017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>26151821</td>\n",
       "      <td>Nat Commun</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/26151821</td>\n",
       "      <td>Genome-wide association study of colorectal ca...</td>\n",
       "      <td>Colorectal cancer</td>\n",
       "      <td>CUX2, BRAP, ACAD10, SH2B3</td>\n",
       "      <td>ATXN2, SH2B3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000204842, ENSG00000111252</td>\n",
       "      <td>rs3184504-C</td>\n",
       "      <td>rs3184504</td>\n",
       "      <td>3184504</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0005842</td>\n",
       "      <td>GCST003017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>26151821</td>\n",
       "      <td>Nat Commun</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/26151821</td>\n",
       "      <td>Genome-wide association study of colorectal ca...</td>\n",
       "      <td>Colorectal cancer</td>\n",
       "      <td>NR</td>\n",
       "      <td>DCBLD1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000164465</td>\n",
       "      <td>rs4946260-T</td>\n",
       "      <td>rs4946260</td>\n",
       "      <td>4946260</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0005842</td>\n",
       "      <td>GCST003017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8242</th>\n",
       "      <td>34930913</td>\n",
       "      <td>Cell Discov</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/34930913</td>\n",
       "      <td>Genome-wide association and functional interro...</td>\n",
       "      <td>ovarian cancer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENPP7P1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000249188</td>\n",
       "      <td>rs142897723-GA</td>\n",
       "      <td>rs142897723</td>\n",
       "      <td>57874171</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>overall survival, ovarian carcinoma</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0000638, http://w...</td>\n",
       "      <td>GCST90102289</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8243</th>\n",
       "      <td>35658861</td>\n",
       "      <td>BMC Med</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/35658861</td>\n",
       "      <td>Associations of genetic risk, BMI trajectories...</td>\n",
       "      <td>lung cancer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SLC16A7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000118596</td>\n",
       "      <td>rs79297227-C</td>\n",
       "      <td>rs79297227</td>\n",
       "      <td>79297227</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>longitudinal BMI measurement, non-small cell l...</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0005937, http://w...</td>\n",
       "      <td>GCST90132908</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8244</th>\n",
       "      <td>35658861</td>\n",
       "      <td>BMC Med</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/35658861</td>\n",
       "      <td>Associations of genetic risk, BMI trajectories...</td>\n",
       "      <td>lung cancer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LINC01811</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000226320</td>\n",
       "      <td>rs2336652-A</td>\n",
       "      <td>rs2336652</td>\n",
       "      <td>2336652</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>longitudinal BMI measurement, non-small cell l...</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0005937, http://w...</td>\n",
       "      <td>GCST90132908</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8245</th>\n",
       "      <td>35658861</td>\n",
       "      <td>BMC Med</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/35658861</td>\n",
       "      <td>Associations of genetic risk, BMI trajectories...</td>\n",
       "      <td>lung cancer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CACNA1A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000141837</td>\n",
       "      <td>rs16018-G</td>\n",
       "      <td>rs16018</td>\n",
       "      <td>16018</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>longitudinal BMI measurement, non-small cell l...</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0005937, http://w...</td>\n",
       "      <td>GCST90132908</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8246</th>\n",
       "      <td>35658861</td>\n",
       "      <td>BMC Med</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/35658861</td>\n",
       "      <td>Associations of genetic risk, BMI trajectories...</td>\n",
       "      <td>lung cancer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SLC37A3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000157800</td>\n",
       "      <td>rs4726760-T</td>\n",
       "      <td>rs4726760</td>\n",
       "      <td>4726760</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>longitudinal BMI measurement, non-small cell l...</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0005937, http://w...</td>\n",
       "      <td>GCST90132908</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8247 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      PUBMEDID      JOURNAL                                  LINK  \\\n",
       "0     26151821   Nat Commun  www.ncbi.nlm.nih.gov/pubmed/26151821   \n",
       "1     26151821   Nat Commun  www.ncbi.nlm.nih.gov/pubmed/26151821   \n",
       "2     26151821   Nat Commun  www.ncbi.nlm.nih.gov/pubmed/26151821   \n",
       "3     26151821   Nat Commun  www.ncbi.nlm.nih.gov/pubmed/26151821   \n",
       "4     26151821   Nat Commun  www.ncbi.nlm.nih.gov/pubmed/26151821   \n",
       "...        ...          ...                                   ...   \n",
       "8242  34930913  Cell Discov  www.ncbi.nlm.nih.gov/pubmed/34930913   \n",
       "8243  35658861      BMC Med  www.ncbi.nlm.nih.gov/pubmed/35658861   \n",
       "8244  35658861      BMC Med  www.ncbi.nlm.nih.gov/pubmed/35658861   \n",
       "8245  35658861      BMC Med  www.ncbi.nlm.nih.gov/pubmed/35658861   \n",
       "8246  35658861      BMC Med  www.ncbi.nlm.nih.gov/pubmed/35658861   \n",
       "\n",
       "                                                  STUDY      DISEASE/TRAIT  \\\n",
       "0     Genome-wide association study of colorectal ca...  Colorectal cancer   \n",
       "1     Genome-wide association study of colorectal ca...  Colorectal cancer   \n",
       "2     Genome-wide association study of colorectal ca...  Colorectal cancer   \n",
       "3     Genome-wide association study of colorectal ca...  Colorectal cancer   \n",
       "4     Genome-wide association study of colorectal ca...  Colorectal cancer   \n",
       "...                                                 ...                ...   \n",
       "8242  Genome-wide association and functional interro...     ovarian cancer   \n",
       "8243  Associations of genetic risk, BMI trajectories...        lung cancer   \n",
       "8244  Associations of genetic risk, BMI trajectories...        lung cancer   \n",
       "8245  Associations of genetic risk, BMI trajectories...        lung cancer   \n",
       "8246  Associations of genetic risk, BMI trajectories...        lung cancer   \n",
       "\n",
       "                          REPORTED GENE(S)        MAPPED_GENE  \\\n",
       "0                          SLC25A26, LRIG1              LRIG1   \n",
       "1                                     NOS1               NOS1   \n",
       "2     ENTPD7, COX15, CUTC, ABCC2, SLC25A28  NKX2-3 - SLC25A28   \n",
       "3                CUX2, BRAP, ACAD10, SH2B3       ATXN2, SH2B3   \n",
       "4                                       NR             DCBLD1   \n",
       "...                                    ...                ...   \n",
       "8242                                   NaN            ENPP7P1   \n",
       "8243                                   NaN            SLC16A7   \n",
       "8244                                   NaN          LINC01811   \n",
       "8245                                   NaN            CACNA1A   \n",
       "8246                                   NaN            SLC37A3   \n",
       "\n",
       "     UPSTREAM_GENE_ID DOWNSTREAM_GENE_ID                      SNP_GENE_IDS  \\\n",
       "0                 NaN                NaN                   ENSG00000144749   \n",
       "1                 NaN                NaN                   ENSG00000089250   \n",
       "2     ENSG00000119919    ENSG00000155287                               NaN   \n",
       "3                 NaN                NaN  ENSG00000204842, ENSG00000111252   \n",
       "4                 NaN                NaN                   ENSG00000164465   \n",
       "...               ...                ...                               ...   \n",
       "8242              NaN                NaN                   ENSG00000249188   \n",
       "8243              NaN                NaN                   ENSG00000118596   \n",
       "8244              NaN                NaN                   ENSG00000226320   \n",
       "8245              NaN                NaN                   ENSG00000141837   \n",
       "8246              NaN                NaN                   ENSG00000157800   \n",
       "\n",
       "     STRONGEST SNP-RISK ALLELE         SNPS SNP_ID_CURRENT  \\\n",
       "0                   rs812481-G     rs812481         812481   \n",
       "1                 rs73208120-G   rs73208120       73208120   \n",
       "2                 rs11190164-G   rs11190164       11190164   \n",
       "3                  rs3184504-C    rs3184504        3184504   \n",
       "4                  rs4946260-T    rs4946260        4946260   \n",
       "...                        ...          ...            ...   \n",
       "8242            rs142897723-GA  rs142897723       57874171   \n",
       "8243              rs79297227-C   rs79297227       79297227   \n",
       "8244               rs2336652-A    rs2336652        2336652   \n",
       "8245                 rs16018-G      rs16018          16018   \n",
       "8246               rs4726760-T    rs4726760        4726760   \n",
       "\n",
       "                 CONTEXT CNV  \\\n",
       "0         intron_variant   N   \n",
       "1         intron_variant   N   \n",
       "2     intergenic_variant   N   \n",
       "3       missense_variant   N   \n",
       "4         intron_variant   N   \n",
       "...                  ...  ..   \n",
       "8242      intron_variant   N   \n",
       "8243      intron_variant   N   \n",
       "8244      intron_variant   N   \n",
       "8245      intron_variant   N   \n",
       "8246      intron_variant   N   \n",
       "\n",
       "                                           MAPPED_TRAIT  \\\n",
       "0                                     colorectal cancer   \n",
       "1                                     colorectal cancer   \n",
       "2                                     colorectal cancer   \n",
       "3                                     colorectal cancer   \n",
       "4                                     colorectal cancer   \n",
       "...                                                 ...   \n",
       "8242                overall survival, ovarian carcinoma   \n",
       "8243  longitudinal BMI measurement, non-small cell l...   \n",
       "8244  longitudinal BMI measurement, non-small cell l...   \n",
       "8245  longitudinal BMI measurement, non-small cell l...   \n",
       "8246  longitudinal BMI measurement, non-small cell l...   \n",
       "\n",
       "                                       MAPPED_TRAIT_URI STUDY ACCESSION  \n",
       "0                  http://www.ebi.ac.uk/efo/EFO_0005842      GCST003017  \n",
       "1                  http://www.ebi.ac.uk/efo/EFO_0005842      GCST003017  \n",
       "2                  http://www.ebi.ac.uk/efo/EFO_0005842      GCST003017  \n",
       "3                  http://www.ebi.ac.uk/efo/EFO_0005842      GCST003017  \n",
       "4                  http://www.ebi.ac.uk/efo/EFO_0005842      GCST003017  \n",
       "...                                                 ...             ...  \n",
       "8242  http://www.ebi.ac.uk/efo/EFO_0000638, http://w...    GCST90102289  \n",
       "8243  http://www.ebi.ac.uk/efo/EFO_0005937, http://w...    GCST90132908  \n",
       "8244  http://www.ebi.ac.uk/efo/EFO_0005937, http://w...    GCST90132908  \n",
       "8245  http://www.ebi.ac.uk/efo/EFO_0005937, http://w...    GCST90132908  \n",
       "8246  http://www.ebi.ac.uk/efo/EFO_0005937, http://w...    GCST90132908  \n",
       "\n",
       "[8247 rows x 18 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read in gwas raw data\n",
    "gwas_raw = helper.read_in_data('../data/gwas_cancer_revised.csv', 'csv')\n",
    "gwas_raw"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PUBMEDID</th>\n",
       "      <th>JOURNAL</th>\n",
       "      <th>LINK</th>\n",
       "      <th>STUDY</th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>REPORTED GENE(S)</th>\n",
       "      <th>UPSTREAM_GENE_ID</th>\n",
       "      <th>DOWNSTREAM_GENE_ID</th>\n",
       "      <th>SNP_GENE_IDS</th>\n",
       "      <th>STRONGEST SNP-RISK ALLELE</th>\n",
       "      <th>SNPS</th>\n",
       "      <th>SNP_ID_CURRENT</th>\n",
       "      <th>CONTEXT</th>\n",
       "      <th>CNV</th>\n",
       "      <th>MAPPED_TRAIT</th>\n",
       "      <th>MAPPED_TRAIT_URI</th>\n",
       "      <th>STUDY ACCESSION</th>\n",
       "      <th>biomarker</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>26151821</td>\n",
       "      <td>Nat Commun</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/26151821</td>\n",
       "      <td>Genome-wide association study of colorectal ca...</td>\n",
       "      <td>Colorectal cancer</td>\n",
       "      <td>SLC25A26, LRIG1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000144749</td>\n",
       "      <td>rs812481-G</td>\n",
       "      <td>rs812481</td>\n",
       "      <td>812481</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0005842</td>\n",
       "      <td>GCST003017</td>\n",
       "      <td>LRIG1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>26151821</td>\n",
       "      <td>Nat Commun</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/26151821</td>\n",
       "      <td>Genome-wide association study of colorectal ca...</td>\n",
       "      <td>Colorectal cancer</td>\n",
       "      <td>NOS1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000089250</td>\n",
       "      <td>rs73208120-G</td>\n",
       "      <td>rs73208120</td>\n",
       "      <td>73208120</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0005842</td>\n",
       "      <td>GCST003017</td>\n",
       "      <td>NOS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>26151821</td>\n",
       "      <td>Nat Commun</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/26151821</td>\n",
       "      <td>Genome-wide association study of colorectal ca...</td>\n",
       "      <td>Colorectal cancer</td>\n",
       "      <td>ENTPD7, COX15, CUTC, ABCC2, SLC25A28</td>\n",
       "      <td>ENSG00000119919</td>\n",
       "      <td>ENSG00000155287</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rs11190164-G</td>\n",
       "      <td>rs11190164</td>\n",
       "      <td>11190164</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0005842</td>\n",
       "      <td>GCST003017</td>\n",
       "      <td>NKX2-3 - SLC25A28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>26151821</td>\n",
       "      <td>Nat Commun</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/26151821</td>\n",
       "      <td>Genome-wide association study of colorectal ca...</td>\n",
       "      <td>Colorectal cancer</td>\n",
       "      <td>CUX2, BRAP, ACAD10, SH2B3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000204842, ENSG00000111252</td>\n",
       "      <td>rs3184504-C</td>\n",
       "      <td>rs3184504</td>\n",
       "      <td>3184504</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0005842</td>\n",
       "      <td>GCST003017</td>\n",
       "      <td>ATXN2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>26151821</td>\n",
       "      <td>Nat Commun</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/26151821</td>\n",
       "      <td>Genome-wide association study of colorectal ca...</td>\n",
       "      <td>Colorectal cancer</td>\n",
       "      <td>CUX2, BRAP, ACAD10, SH2B3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000204842, ENSG00000111252</td>\n",
       "      <td>rs3184504-C</td>\n",
       "      <td>rs3184504</td>\n",
       "      <td>3184504</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0005842</td>\n",
       "      <td>GCST003017</td>\n",
       "      <td>SH2B3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9743</th>\n",
       "      <td>34930913</td>\n",
       "      <td>Cell Discov</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/34930913</td>\n",
       "      <td>Genome-wide association and functional interro...</td>\n",
       "      <td>ovarian cancer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000249188</td>\n",
       "      <td>rs142897723-GA</td>\n",
       "      <td>rs142897723</td>\n",
       "      <td>57874171</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>overall survival, ovarian carcinoma</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0000638, http://w...</td>\n",
       "      <td>GCST90102289</td>\n",
       "      <td>ENPP7P1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9744</th>\n",
       "      <td>35658861</td>\n",
       "      <td>BMC Med</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/35658861</td>\n",
       "      <td>Associations of genetic risk, BMI trajectories...</td>\n",
       "      <td>lung cancer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000118596</td>\n",
       "      <td>rs79297227-C</td>\n",
       "      <td>rs79297227</td>\n",
       "      <td>79297227</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>longitudinal BMI measurement, non-small cell l...</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0005937, http://w...</td>\n",
       "      <td>GCST90132908</td>\n",
       "      <td>SLC16A7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9745</th>\n",
       "      <td>35658861</td>\n",
       "      <td>BMC Med</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/35658861</td>\n",
       "      <td>Associations of genetic risk, BMI trajectories...</td>\n",
       "      <td>lung cancer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000226320</td>\n",
       "      <td>rs2336652-A</td>\n",
       "      <td>rs2336652</td>\n",
       "      <td>2336652</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>longitudinal BMI measurement, non-small cell l...</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0005937, http://w...</td>\n",
       "      <td>GCST90132908</td>\n",
       "      <td>LINC01811</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9746</th>\n",
       "      <td>35658861</td>\n",
       "      <td>BMC Med</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/35658861</td>\n",
       "      <td>Associations of genetic risk, BMI trajectories...</td>\n",
       "      <td>lung cancer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000141837</td>\n",
       "      <td>rs16018-G</td>\n",
       "      <td>rs16018</td>\n",
       "      <td>16018</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>longitudinal BMI measurement, non-small cell l...</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0005937, http://w...</td>\n",
       "      <td>GCST90132908</td>\n",
       "      <td>CACNA1A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9747</th>\n",
       "      <td>35658861</td>\n",
       "      <td>BMC Med</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/35658861</td>\n",
       "      <td>Associations of genetic risk, BMI trajectories...</td>\n",
       "      <td>lung cancer</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000157800</td>\n",
       "      <td>rs4726760-T</td>\n",
       "      <td>rs4726760</td>\n",
       "      <td>4726760</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>N</td>\n",
       "      <td>longitudinal BMI measurement, non-small cell l...</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0005937, http://w...</td>\n",
       "      <td>GCST90132908</td>\n",
       "      <td>SLC37A3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9748 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      PUBMEDID      JOURNAL                                  LINK  \\\n",
       "0     26151821   Nat Commun  www.ncbi.nlm.nih.gov/pubmed/26151821   \n",
       "1     26151821   Nat Commun  www.ncbi.nlm.nih.gov/pubmed/26151821   \n",
       "2     26151821   Nat Commun  www.ncbi.nlm.nih.gov/pubmed/26151821   \n",
       "3     26151821   Nat Commun  www.ncbi.nlm.nih.gov/pubmed/26151821   \n",
       "4     26151821   Nat Commun  www.ncbi.nlm.nih.gov/pubmed/26151821   \n",
       "...        ...          ...                                   ...   \n",
       "9743  34930913  Cell Discov  www.ncbi.nlm.nih.gov/pubmed/34930913   \n",
       "9744  35658861      BMC Med  www.ncbi.nlm.nih.gov/pubmed/35658861   \n",
       "9745  35658861      BMC Med  www.ncbi.nlm.nih.gov/pubmed/35658861   \n",
       "9746  35658861      BMC Med  www.ncbi.nlm.nih.gov/pubmed/35658861   \n",
       "9747  35658861      BMC Med  www.ncbi.nlm.nih.gov/pubmed/35658861   \n",
       "\n",
       "                                                  STUDY      DISEASE/TRAIT  \\\n",
       "0     Genome-wide association study of colorectal ca...  Colorectal cancer   \n",
       "1     Genome-wide association study of colorectal ca...  Colorectal cancer   \n",
       "2     Genome-wide association study of colorectal ca...  Colorectal cancer   \n",
       "3     Genome-wide association study of colorectal ca...  Colorectal cancer   \n",
       "4     Genome-wide association study of colorectal ca...  Colorectal cancer   \n",
       "...                                                 ...                ...   \n",
       "9743  Genome-wide association and functional interro...     ovarian cancer   \n",
       "9744  Associations of genetic risk, BMI trajectories...        lung cancer   \n",
       "9745  Associations of genetic risk, BMI trajectories...        lung cancer   \n",
       "9746  Associations of genetic risk, BMI trajectories...        lung cancer   \n",
       "9747  Associations of genetic risk, BMI trajectories...        lung cancer   \n",
       "\n",
       "                          REPORTED GENE(S) UPSTREAM_GENE_ID  \\\n",
       "0                          SLC25A26, LRIG1              NaN   \n",
       "1                                     NOS1              NaN   \n",
       "2     ENTPD7, COX15, CUTC, ABCC2, SLC25A28  ENSG00000119919   \n",
       "3                CUX2, BRAP, ACAD10, SH2B3              NaN   \n",
       "4                CUX2, BRAP, ACAD10, SH2B3              NaN   \n",
       "...                                    ...              ...   \n",
       "9743                                   NaN              NaN   \n",
       "9744                                   NaN              NaN   \n",
       "9745                                   NaN              NaN   \n",
       "9746                                   NaN              NaN   \n",
       "9747                                   NaN              NaN   \n",
       "\n",
       "     DOWNSTREAM_GENE_ID                      SNP_GENE_IDS  \\\n",
       "0                   NaN                   ENSG00000144749   \n",
       "1                   NaN                   ENSG00000089250   \n",
       "2       ENSG00000155287                               NaN   \n",
       "3                   NaN  ENSG00000204842, ENSG00000111252   \n",
       "4                   NaN  ENSG00000204842, ENSG00000111252   \n",
       "...                 ...                               ...   \n",
       "9743                NaN                   ENSG00000249188   \n",
       "9744                NaN                   ENSG00000118596   \n",
       "9745                NaN                   ENSG00000226320   \n",
       "9746                NaN                   ENSG00000141837   \n",
       "9747                NaN                   ENSG00000157800   \n",
       "\n",
       "     STRONGEST SNP-RISK ALLELE         SNPS SNP_ID_CURRENT  \\\n",
       "0                   rs812481-G     rs812481         812481   \n",
       "1                 rs73208120-G   rs73208120       73208120   \n",
       "2                 rs11190164-G   rs11190164       11190164   \n",
       "3                  rs3184504-C    rs3184504        3184504   \n",
       "4                  rs3184504-C    rs3184504        3184504   \n",
       "...                        ...          ...            ...   \n",
       "9743            rs142897723-GA  rs142897723       57874171   \n",
       "9744              rs79297227-C   rs79297227       79297227   \n",
       "9745               rs2336652-A    rs2336652        2336652   \n",
       "9746                 rs16018-G      rs16018          16018   \n",
       "9747               rs4726760-T    rs4726760        4726760   \n",
       "\n",
       "                 CONTEXT CNV  \\\n",
       "0         intron_variant   N   \n",
       "1         intron_variant   N   \n",
       "2     intergenic_variant   N   \n",
       "3       missense_variant   N   \n",
       "4       missense_variant   N   \n",
       "...                  ...  ..   \n",
       "9743      intron_variant   N   \n",
       "9744      intron_variant   N   \n",
       "9745      intron_variant   N   \n",
       "9746      intron_variant   N   \n",
       "9747      intron_variant   N   \n",
       "\n",
       "                                           MAPPED_TRAIT  \\\n",
       "0                                     colorectal cancer   \n",
       "1                                     colorectal cancer   \n",
       "2                                     colorectal cancer   \n",
       "3                                     colorectal cancer   \n",
       "4                                     colorectal cancer   \n",
       "...                                                 ...   \n",
       "9743                overall survival, ovarian carcinoma   \n",
       "9744  longitudinal BMI measurement, non-small cell l...   \n",
       "9745  longitudinal BMI measurement, non-small cell l...   \n",
       "9746  longitudinal BMI measurement, non-small cell l...   \n",
       "9747  longitudinal BMI measurement, non-small cell l...   \n",
       "\n",
       "                                       MAPPED_TRAIT_URI STUDY ACCESSION  \\\n",
       "0                  http://www.ebi.ac.uk/efo/EFO_0005842      GCST003017   \n",
       "1                  http://www.ebi.ac.uk/efo/EFO_0005842      GCST003017   \n",
       "2                  http://www.ebi.ac.uk/efo/EFO_0005842      GCST003017   \n",
       "3                  http://www.ebi.ac.uk/efo/EFO_0005842      GCST003017   \n",
       "4                  http://www.ebi.ac.uk/efo/EFO_0005842      GCST003017   \n",
       "...                                                 ...             ...   \n",
       "9743  http://www.ebi.ac.uk/efo/EFO_0000638, http://w...    GCST90102289   \n",
       "9744  http://www.ebi.ac.uk/efo/EFO_0005937, http://w...    GCST90132908   \n",
       "9745  http://www.ebi.ac.uk/efo/EFO_0005937, http://w...    GCST90132908   \n",
       "9746  http://www.ebi.ac.uk/efo/EFO_0005937, http://w...    GCST90132908   \n",
       "9747  http://www.ebi.ac.uk/efo/EFO_0005937, http://w...    GCST90132908   \n",
       "\n",
       "              biomarker  \n",
       "0                 LRIG1  \n",
       "1                  NOS1  \n",
       "2     NKX2-3 - SLC25A28  \n",
       "3                 ATXN2  \n",
       "4                 SH2B3  \n",
       "...                 ...  \n",
       "9743            ENPP7P1  \n",
       "9744            SLC16A7  \n",
       "9745          LINC01811  \n",
       "9746            CACNA1A  \n",
       "9747            SLC37A3  \n",
       "\n",
       "[9748 rows x 18 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# drop columns that have 'NO MAPPED GENES' in the MAPPED_GENE column \n",
    "gwas_df = helper.split_col_on_delim(df = gwas_raw, split_column = 'MAPPED_GENE', new_col_name = 'biomarker', delim = ',')\n",
    "valid_mapped_gene_condition = ~gwas_df['biomarker'].str.contains('no mapped genes', case = False, na = False)\n",
    "gwas_df = gwas_df[valid_mapped_gene_condition]\n",
    "gwas_df.reset_index(drop = True, inplace = True)\n",
    "gwas_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(9767, 18)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def split_x_rows(row):\n",
    "    ''' Split x formatted rows and the corresponding rs id's. \n",
    "    '''\n",
    "    biomarkers = row['biomarker'].split(' x ')\n",
    "    snps = row['SNPS'].split(' x ')\n",
    "\n",
    "    new_row = row.copy()\n",
    "    row['biomarker'] = biomarkers[0]\n",
    "    row['SNPS'] = snps[0]\n",
    "    new_row['biomarker'] = biomarkers[1]\n",
    "    new_row['SNPS'] = snps[1]\n",
    "    return pd.DataFrame([row, new_row])\n",
    "\n",
    "# extract rows with ' x ' formatted genes\n",
    "x_mask = gwas_df['biomarker'].str.contains(' x ', case = False, na = False)\n",
    "x_rows = gwas_df[x_mask]\n",
    "\n",
    "# generate new formatted ' x ' rows \n",
    "new_x_rows = x_rows.apply(split_x_rows, axis = 1).tolist()\n",
    "x_split_rows = pd.concat(new_x_rows, ignore_index = True)\n",
    "\n",
    "# put the newly formatted x rows back into the dataset \n",
    "gwas_df = gwas_df[~x_mask]\n",
    "gwas_df = pd.concat([gwas_df, x_split_rows])\n",
    "gwas_df.reset_index(drop = True, inplace = True)\n",
    "gwas_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PUBMEDID</th>\n",
       "      <th>JOURNAL</th>\n",
       "      <th>LINK</th>\n",
       "      <th>STUDY</th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>REPORTED GENE(S)</th>\n",
       "      <th>UPSTREAM_GENE_ID</th>\n",
       "      <th>DOWNSTREAM_GENE_ID</th>\n",
       "      <th>SNP_GENE_IDS</th>\n",
       "      <th>STRONGEST SNP-RISK ALLELE</th>\n",
       "      <th>SNPS</th>\n",
       "      <th>SNP_ID_CURRENT</th>\n",
       "      <th>CONTEXT</th>\n",
       "      <th>CNV</th>\n",
       "      <th>MAPPED_TRAIT</th>\n",
       "      <th>MAPPED_TRAIT_URI</th>\n",
       "      <th>STUDY ACCESSION</th>\n",
       "      <th>biomarker</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [PUBMEDID, JOURNAL, LINK, STUDY, DISEASE/TRAIT, REPORTED GENE(S), UPSTREAM_GENE_ID, DOWNSTREAM_GENE_ID, SNP_GENE_IDS, STRONGEST SNP-RISK ALLELE, SNPS, SNP_ID_CURRENT, CONTEXT, CNV, MAPPED_TRAIT, MAPPED_TRAIT_URI, STUDY ACCESSION, biomarker]\n",
       "Index: []"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# drop rows that have a ';' in the mapped gene data\n",
    "gwas_df.drop(index = list(gwas_df[gwas_df['biomarker'].str.contains(';')].index), inplace = True)\n",
    "gwas_df.reset_index(drop = True, inplace = True)\n",
    "gwas_df[gwas_df['biomarker'].str.contains(';')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PUBMEDID</th>\n",
       "      <th>JOURNAL</th>\n",
       "      <th>LINK</th>\n",
       "      <th>STUDY</th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>REPORTED GENE(S)</th>\n",
       "      <th>UPSTREAM_GENE_ID</th>\n",
       "      <th>DOWNSTREAM_GENE_ID</th>\n",
       "      <th>SNP_GENE_IDS</th>\n",
       "      <th>STRONGEST SNP-RISK ALLELE</th>\n",
       "      <th>SNPS</th>\n",
       "      <th>SNP_ID_CURRENT</th>\n",
       "      <th>CONTEXT</th>\n",
       "      <th>CNV</th>\n",
       "      <th>MAPPED_TRAIT</th>\n",
       "      <th>MAPPED_TRAIT_URI</th>\n",
       "      <th>STUDY ACCESSION</th>\n",
       "      <th>biomarker</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [PUBMEDID, JOURNAL, LINK, STUDY, DISEASE/TRAIT, REPORTED GENE(S), UPSTREAM_GENE_ID, DOWNSTREAM_GENE_ID, SNP_GENE_IDS, STRONGEST SNP-RISK ALLELE, SNPS, SNP_ID_CURRENT, CONTEXT, CNV, MAPPED_TRAIT, MAPPED_TRAIT_URI, STUDY ACCESSION, biomarker]\n",
       "Index: []"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# drop rows that have a range listed in the mapped gene data \n",
    "gwas_df.drop(index = list(gwas_df[gwas_df['biomarker'].str.contains(' - ')].index), inplace = True)\n",
    "gwas_df.reset_index(drop = True, inplace = True)\n",
    "gwas_df[gwas_df['biomarker'].str.contains(' - ')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(7125, 18)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# split on the disease column when a list is provided\n",
    "gwas_df = helper.split_col_on_delim(df = gwas_df, split_column = 'DISEASE/TRAIT', new_col_name = 'disease', delim = ',')\n",
    "gwas_df.reset_index(drop = True, inplace = True)\n",
    "gwas_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PUBMEDID</th>\n",
       "      <th>JOURNAL</th>\n",
       "      <th>LINK</th>\n",
       "      <th>STUDY</th>\n",
       "      <th>REPORTED GENE(S)</th>\n",
       "      <th>UPSTREAM_GENE_ID</th>\n",
       "      <th>DOWNSTREAM_GENE_ID</th>\n",
       "      <th>SNP_GENE_IDS</th>\n",
       "      <th>STRONGEST SNP-RISK ALLELE</th>\n",
       "      <th>SNPS</th>\n",
       "      <th>SNP_ID_CURRENT</th>\n",
       "      <th>CONTEXT</th>\n",
       "      <th>CNV</th>\n",
       "      <th>MAPPED_TRAIT</th>\n",
       "      <th>MAPPED_TRAIT_URI</th>\n",
       "      <th>STUDY ACCESSION</th>\n",
       "      <th>biomarker</th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [PUBMEDID, JOURNAL, LINK, STUDY, REPORTED GENE(S), UPSTREAM_GENE_ID, DOWNSTREAM_GENE_ID, SNP_GENE_IDS, STRONGEST SNP-RISK ALLELE, SNPS, SNP_ID_CURRENT, CONTEXT, CNV, MAPPED_TRAIT, MAPPED_TRAIT_URI, STUDY ACCESSION, biomarker, disease]\n",
       "Index: []"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# drop rows that have x formatted SNPS values \n",
    "gwas_df.drop(index = list(gwas_df[gwas_df['SNPS'].str.contains(' x ')].index), inplace = True)\n",
    "gwas_df.reset_index(drop = True, inplace = True)\n",
    "gwas_df[gwas_df['SNPS'].str.contains(' x ')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biomarker_id</th>\n",
       "      <th>main_x_ref</th>\n",
       "      <th>assessed_biomarker_entity</th>\n",
       "      <th>biomarker_status</th>\n",
       "      <th>best_biomarker_type</th>\n",
       "      <th>specimen_type</th>\n",
       "      <th>loinc_code</th>\n",
       "      <th>condition_name</th>\n",
       "      <th>assessed_entity_type</th>\n",
       "      <th>evidence_source</th>\n",
       "      <th>notes</th>\n",
       "      <th>rs_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>disease</th>\n",
       "      <th>uniprot</th>\n",
       "      <th>name</th>\n",
       "      <th>system</th>\n",
       "      <th>doid</th>\n",
       "      <th>mutation</th>\n",
       "      <th>variation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>dbSNP:rs812481</td>\n",
       "      <td>rs812481 mutation in leucine-rich repeat prote...</td>\n",
       "      <td>presence of</td>\n",
       "      <td>risk_biomarker</td>\n",
       "      <td>nan</td>\n",
       "      <td>nan</td>\n",
       "      <td>colorectal cancer (DOID:9256.0)</td>\n",
       "      <td>gene</td>\n",
       "      <td>GWAS Catalog|https://www.ebi.ac.uk/gwas/search...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>812481</td>\n",
       "      <td>LRIG1</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>Q5XWD3</td>\n",
       "      <td>Leucine-rich repeat protein LRIG1</td>\n",
       "      <td>nan</td>\n",
       "      <td>9256.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>dbSNP:rs73208120</td>\n",
       "      <td>rs73208120 mutation in alternative protein nos...</td>\n",
       "      <td>presence of</td>\n",
       "      <td>risk_biomarker</td>\n",
       "      <td>nan</td>\n",
       "      <td>nan</td>\n",
       "      <td>colorectal cancer (DOID:9256.0)</td>\n",
       "      <td>gene</td>\n",
       "      <td>GWAS Catalog|https://www.ebi.ac.uk/gwas/search...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>73208120</td>\n",
       "      <td>NOS1</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>L8E6X1</td>\n",
       "      <td>Alternative protein NOS1</td>\n",
       "      <td>nan</td>\n",
       "      <td>9256.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>dbSNP:rs3184504</td>\n",
       "      <td>rs3184504 mutation in ataxin-2 (ATXN2)</td>\n",
       "      <td>presence of</td>\n",
       "      <td>risk_biomarker</td>\n",
       "      <td>nan</td>\n",
       "      <td>nan</td>\n",
       "      <td>colorectal cancer (DOID:9256.0)</td>\n",
       "      <td>gene</td>\n",
       "      <td>GWAS Catalog|https://www.ebi.ac.uk/gwas/search...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>3184504</td>\n",
       "      <td>ATXN2</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>V9GY86</td>\n",
       "      <td>Ataxin-2</td>\n",
       "      <td>nan</td>\n",
       "      <td>9256.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>dbSNP:rs3184504</td>\n",
       "      <td>rs3184504 mutation in sh2b adapter protein 3 (...</td>\n",
       "      <td>presence of</td>\n",
       "      <td>risk_biomarker</td>\n",
       "      <td>nan</td>\n",
       "      <td>nan</td>\n",
       "      <td>colorectal cancer (DOID:9256.0)</td>\n",
       "      <td>gene</td>\n",
       "      <td>GWAS Catalog|https://www.ebi.ac.uk/gwas/search...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>3184504</td>\n",
       "      <td>SH2B3</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>R4GN84</td>\n",
       "      <td>SH2B adapter protein 3</td>\n",
       "      <td>nan</td>\n",
       "      <td>9256.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>dbSNP:rs4946260</td>\n",
       "      <td>rs4946260 mutation in discoidin, cub and lccl ...</td>\n",
       "      <td>presence of</td>\n",
       "      <td>risk_biomarker</td>\n",
       "      <td>nan</td>\n",
       "      <td>nan</td>\n",
       "      <td>colorectal cancer (DOID:9256.0)</td>\n",
       "      <td>gene</td>\n",
       "      <td>GWAS Catalog|https://www.ebi.ac.uk/gwas/search...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>4946260</td>\n",
       "      <td>DCBLD1</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>H0Y4G4</td>\n",
       "      <td>Discoidin, CUB and LCCL domain-containing prot...</td>\n",
       "      <td>nan</td>\n",
       "      <td>9256.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7114</th>\n",
       "      <td>NaN</td>\n",
       "      <td>dbSNP:rs8034191</td>\n",
       "      <td>rs8034191 mutation in hydroxylysine kinase (HYKK)</td>\n",
       "      <td>presence of</td>\n",
       "      <td>risk_biomarker</td>\n",
       "      <td>nan</td>\n",
       "      <td>nan</td>\n",
       "      <td>lung cancer (DOID:1324.0)</td>\n",
       "      <td>gene</td>\n",
       "      <td>GWAS Catalog|https://www.ebi.ac.uk/gwas/search...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>8034191</td>\n",
       "      <td>HYKK</td>\n",
       "      <td>lung cancer</td>\n",
       "      <td>A0A0C4DGM4</td>\n",
       "      <td>Hydroxylysine kinase</td>\n",
       "      <td>nan</td>\n",
       "      <td>1324.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7115</th>\n",
       "      <td>NaN</td>\n",
       "      <td>dbSNP:rs551610</td>\n",
       "      <td>rs551610 mutation in rims1 protein (RIMS1)</td>\n",
       "      <td>presence of</td>\n",
       "      <td>risk_biomarker</td>\n",
       "      <td>nan</td>\n",
       "      <td>nan</td>\n",
       "      <td>lung cancer (DOID:1324.0)</td>\n",
       "      <td>gene</td>\n",
       "      <td>GWAS Catalog|https://www.ebi.ac.uk/gwas/search...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>551610</td>\n",
       "      <td>RIMS1</td>\n",
       "      <td>lung cancer</td>\n",
       "      <td>Q3ZCW0</td>\n",
       "      <td>RIMS1 protein</td>\n",
       "      <td>nan</td>\n",
       "      <td>1324.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7116</th>\n",
       "      <td>NaN</td>\n",
       "      <td>dbSNP:rs1051730</td>\n",
       "      <td>rs1051730 mutation in neuronal nicotinic acety...</td>\n",
       "      <td>presence of</td>\n",
       "      <td>risk_biomarker</td>\n",
       "      <td>nan</td>\n",
       "      <td>nan</td>\n",
       "      <td>lung cancer (DOID:1324.0)</td>\n",
       "      <td>gene</td>\n",
       "      <td>GWAS Catalog|https://www.ebi.ac.uk/gwas/search...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>1051730</td>\n",
       "      <td>CHRNA3</td>\n",
       "      <td>lung cancer</td>\n",
       "      <td>Q6EWN2</td>\n",
       "      <td>Neuronal nicotinic acetylcholine receptor, alp...</td>\n",
       "      <td>nan</td>\n",
       "      <td>1324.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7117</th>\n",
       "      <td>NaN</td>\n",
       "      <td>dbSNP:rs8034191</td>\n",
       "      <td>rs8034191 mutation in hydroxylysine kinase (HYKK)</td>\n",
       "      <td>presence of</td>\n",
       "      <td>risk_biomarker</td>\n",
       "      <td>nan</td>\n",
       "      <td>nan</td>\n",
       "      <td>lung cancer (DOID:1324.0)</td>\n",
       "      <td>gene</td>\n",
       "      <td>GWAS Catalog|https://www.ebi.ac.uk/gwas/search...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>8034191</td>\n",
       "      <td>HYKK</td>\n",
       "      <td>lung cancer</td>\n",
       "      <td>A0A0C4DGM4</td>\n",
       "      <td>Hydroxylysine kinase</td>\n",
       "      <td>nan</td>\n",
       "      <td>1324.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7118</th>\n",
       "      <td>NaN</td>\n",
       "      <td>dbSNP:rs2744570</td>\n",
       "      <td>rs2744570 mutation in nan (GPLD1)</td>\n",
       "      <td>presence of</td>\n",
       "      <td>risk_biomarker</td>\n",
       "      <td>nan</td>\n",
       "      <td>nan</td>\n",
       "      <td>lung cancer (DOID:1324.0)</td>\n",
       "      <td>gene</td>\n",
       "      <td>GWAS Catalog|https://www.ebi.ac.uk/gwas/search...</td>\n",
       "      <td>GRCh38</td>\n",
       "      <td>2744570</td>\n",
       "      <td>GPLD1</td>\n",
       "      <td>lung cancer</td>\n",
       "      <td>Null</td>\n",
       "      <td>nan</td>\n",
       "      <td>nan</td>\n",
       "      <td>1324.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7119 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     biomarker_id        main_x_ref  \\\n",
       "0             NaN    dbSNP:rs812481   \n",
       "1             NaN  dbSNP:rs73208120   \n",
       "2             NaN   dbSNP:rs3184504   \n",
       "3             NaN   dbSNP:rs3184504   \n",
       "4             NaN   dbSNP:rs4946260   \n",
       "...           ...               ...   \n",
       "7114          NaN   dbSNP:rs8034191   \n",
       "7115          NaN    dbSNP:rs551610   \n",
       "7116          NaN   dbSNP:rs1051730   \n",
       "7117          NaN   dbSNP:rs8034191   \n",
       "7118          NaN   dbSNP:rs2744570   \n",
       "\n",
       "                              assessed_biomarker_entity biomarker_status  \\\n",
       "0     rs812481 mutation in leucine-rich repeat prote...      presence of   \n",
       "1     rs73208120 mutation in alternative protein nos...      presence of   \n",
       "2                rs3184504 mutation in ataxin-2 (ATXN2)      presence of   \n",
       "3     rs3184504 mutation in sh2b adapter protein 3 (...      presence of   \n",
       "4     rs4946260 mutation in discoidin, cub and lccl ...      presence of   \n",
       "...                                                 ...              ...   \n",
       "7114  rs8034191 mutation in hydroxylysine kinase (HYKK)      presence of   \n",
       "7115         rs551610 mutation in rims1 protein (RIMS1)      presence of   \n",
       "7116  rs1051730 mutation in neuronal nicotinic acety...      presence of   \n",
       "7117  rs8034191 mutation in hydroxylysine kinase (HYKK)      presence of   \n",
       "7118                  rs2744570 mutation in nan (GPLD1)      presence of   \n",
       "\n",
       "     best_biomarker_type specimen_type loinc_code  \\\n",
       "0         risk_biomarker           nan        nan   \n",
       "1         risk_biomarker           nan        nan   \n",
       "2         risk_biomarker           nan        nan   \n",
       "3         risk_biomarker           nan        nan   \n",
       "4         risk_biomarker           nan        nan   \n",
       "...                  ...           ...        ...   \n",
       "7114      risk_biomarker           nan        nan   \n",
       "7115      risk_biomarker           nan        nan   \n",
       "7116      risk_biomarker           nan        nan   \n",
       "7117      risk_biomarker           nan        nan   \n",
       "7118      risk_biomarker           nan        nan   \n",
       "\n",
       "                       condition_name assessed_entity_type  \\\n",
       "0     colorectal cancer (DOID:9256.0)                 gene   \n",
       "1     colorectal cancer (DOID:9256.0)                 gene   \n",
       "2     colorectal cancer (DOID:9256.0)                 gene   \n",
       "3     colorectal cancer (DOID:9256.0)                 gene   \n",
       "4     colorectal cancer (DOID:9256.0)                 gene   \n",
       "...                               ...                  ...   \n",
       "7114        lung cancer (DOID:1324.0)                 gene   \n",
       "7115        lung cancer (DOID:1324.0)                 gene   \n",
       "7116        lung cancer (DOID:1324.0)                 gene   \n",
       "7117        lung cancer (DOID:1324.0)                 gene   \n",
       "7118        lung cancer (DOID:1324.0)                 gene   \n",
       "\n",
       "                                        evidence_source   notes     rs_id  \\\n",
       "0     GWAS Catalog|https://www.ebi.ac.uk/gwas/search...  GRCh38    812481   \n",
       "1     GWAS Catalog|https://www.ebi.ac.uk/gwas/search...  GRCh38  73208120   \n",
       "2     GWAS Catalog|https://www.ebi.ac.uk/gwas/search...  GRCh38   3184504   \n",
       "3     GWAS Catalog|https://www.ebi.ac.uk/gwas/search...  GRCh38   3184504   \n",
       "4     GWAS Catalog|https://www.ebi.ac.uk/gwas/search...  GRCh38   4946260   \n",
       "...                                                 ...     ...       ...   \n",
       "7114  GWAS Catalog|https://www.ebi.ac.uk/gwas/search...  GRCh38   8034191   \n",
       "7115  GWAS Catalog|https://www.ebi.ac.uk/gwas/search...  GRCh38    551610   \n",
       "7116  GWAS Catalog|https://www.ebi.ac.uk/gwas/search...  GRCh38   1051730   \n",
       "7117  GWAS Catalog|https://www.ebi.ac.uk/gwas/search...  GRCh38   8034191   \n",
       "7118  GWAS Catalog|https://www.ebi.ac.uk/gwas/search...  GRCh38   2744570   \n",
       "\n",
       "        gene            disease     uniprot  \\\n",
       "0      LRIG1  colorectal cancer      Q5XWD3   \n",
       "1       NOS1  colorectal cancer      L8E6X1   \n",
       "2      ATXN2  colorectal cancer      V9GY86   \n",
       "3      SH2B3  colorectal cancer      R4GN84   \n",
       "4     DCBLD1  colorectal cancer      H0Y4G4   \n",
       "...      ...                ...         ...   \n",
       "7114    HYKK        lung cancer  A0A0C4DGM4   \n",
       "7115   RIMS1        lung cancer      Q3ZCW0   \n",
       "7116  CHRNA3        lung cancer      Q6EWN2   \n",
       "7117    HYKK        lung cancer  A0A0C4DGM4   \n",
       "7118   GPLD1        lung cancer        Null   \n",
       "\n",
       "                                                   name system    doid  \\\n",
       "0                     Leucine-rich repeat protein LRIG1    nan  9256.0   \n",
       "1                              Alternative protein NOS1    nan  9256.0   \n",
       "2                                              Ataxin-2    nan  9256.0   \n",
       "3                                SH2B adapter protein 3    nan  9256.0   \n",
       "4     Discoidin, CUB and LCCL domain-containing prot...    nan  9256.0   \n",
       "...                                                 ...    ...     ...   \n",
       "7114                               Hydroxylysine kinase    nan  1324.0   \n",
       "7115                                      RIMS1 protein    nan  1324.0   \n",
       "7116  Neuronal nicotinic acetylcholine receptor, alp...    nan  1324.0   \n",
       "7117                               Hydroxylysine kinase    nan  1324.0   \n",
       "7118                                                nan    nan  1324.0   \n",
       "\n",
       "     mutation variation  \n",
       "0         NaN       NaN  \n",
       "1         NaN       NaN  \n",
       "2         NaN       NaN  \n",
       "3         NaN       NaN  \n",
       "4         NaN       NaN  \n",
       "...       ...       ...  \n",
       "7114      NaN       NaN  \n",
       "7115      NaN       NaN  \n",
       "7116      NaN       NaN  \n",
       "7117      NaN       NaN  \n",
       "7118      NaN       NaN  \n",
       "\n",
       "[7119 rows x 20 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# gene column\n",
    "gene_series = gwas_df['biomarker']\n",
    "helper.populate_col(source = gene_series, target_col = 'gene')\n",
    "\n",
    "# rs id column \n",
    "rs_id_series = gwas_df['SNPS'].apply(lambda x: x[2:])\n",
    "helper.populate_col(source = rs_id_series, target_col = 'rs_id', dtype = 'Int64')\n",
    "\n",
    "# evidence source column \n",
    "evidence_series = gwas_df['Evidence'] = 'GWAS Catalog|https://www.ebi.ac.uk/gwas/search?query=' + gwas_df['biomarker']\n",
    "helper.populate_col(source = evidence_series, target_col = 'evidence_source')\n",
    "\n",
    "# notes column \n",
    "notes_series = pd.Series('GRCh38', index = helper.biomarker_df.index)\n",
    "helper.populate_col(source = notes_series, target_col = 'notes')\n",
    "\n",
    "# disease name column \n",
    "disease_series = gwas_df['disease'].str.lower() \n",
    "helper.populate_col(source = disease_series, target_col = 'disease')\n",
    "\n",
    "# assessed entity type column \n",
    "assessed_type_series = pd.Series('gene or protein', index = helper.biomarker_df.index)\n",
    "helper.populate_col(source = assessed_type_series, target_col = 'assessed_entity_type')\n",
    "\n",
    "# main_x_ref column \n",
    "helper.map_x_ref(prefix = 'dbSNP:rs')\n",
    "\n",
    "# biomarker status column \n",
    "helper.set_biomarker_status(val = 'presence of')\n",
    "\n",
    "# best biomarker type column\n",
    "helper.set_best_biomarker_type(val = 'risk_biomarker')\n",
    "\n",
    "# assessed entity type column\n",
    "helper.set_assessed_entity_type(val = 'gene')\n",
    "\n",
    "helper.biomarker_df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.4"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
