{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GWAS \n",
    "\n",
    "Original data was pulled from [here](https://www.ebi.ac.uk/gwas/docs/file-downloads) (all associations v1.0) on 2/14/24."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import table_cleaning_functions as tcf "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Preliminary Wrangling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_3545/2555861512.py:2: DtypeWarning: Columns (9,11,12,13,23,26) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  gwas_raw = pd.read_csv('../home/data/raw_data/GlyGen/gwas_raw.tsv', sep='\\t')\n"
     ]
    }
   ],
   "source": [
    "# read in raw data \n",
    "gwas_raw = pd.read_csv('../home/data/raw_data/GlyGen/gwas_raw.tsv', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(571148, 34)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DATE ADDED TO CATALOG</th>\n",
       "      <th>PUBMEDID</th>\n",
       "      <th>FIRST AUTHOR</th>\n",
       "      <th>DATE</th>\n",
       "      <th>JOURNAL</th>\n",
       "      <th>LINK</th>\n",
       "      <th>STUDY</th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>INITIAL SAMPLE SIZE</th>\n",
       "      <th>REPLICATION SAMPLE SIZE</th>\n",
       "      <th>...</th>\n",
       "      <th>CONTEXT</th>\n",
       "      <th>INTERGENIC</th>\n",
       "      <th>RISK ALLELE FREQUENCY</th>\n",
       "      <th>P-VALUE</th>\n",
       "      <th>PVALUE_MLOG</th>\n",
       "      <th>P-VALUE (TEXT)</th>\n",
       "      <th>OR or BETA</th>\n",
       "      <th>95% CI (TEXT)</th>\n",
       "      <th>PLATFORM [SNPS PASSING QC]</th>\n",
       "      <th>CNV</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2017-08-10</td>\n",
       "      <td>28443625</td>\n",
       "      <td>Justice AE</td>\n",
       "      <td>2017-04-26</td>\n",
       "      <td>Nat Commun</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/28443625</td>\n",
       "      <td>Genome-wide meta-analysis of 241,258 adults ac...</td>\n",
       "      <td>BMI in non-smokers</td>\n",
       "      <td>77,279 European ancestry women, 47,280 Europea...</td>\n",
       "      <td>16,011 European ancestry women, 17,912 Europea...</td>\n",
       "      <td>...</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.847</td>\n",
       "      <td>5.000000e-08</td>\n",
       "      <td>7.301030</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0230</td>\n",
       "      <td>[0.015-0.031] kg/m2 increase</td>\n",
       "      <td>Affymetrix, Illumina, Perlegen [up to 2800000]...</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2017-08-10</td>\n",
       "      <td>28443625</td>\n",
       "      <td>Justice AE</td>\n",
       "      <td>2017-04-26</td>\n",
       "      <td>Nat Commun</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/28443625</td>\n",
       "      <td>Genome-wide meta-analysis of 241,258 adults ac...</td>\n",
       "      <td>BMI in non-smokers</td>\n",
       "      <td>77,279 European ancestry women, 47,280 Europea...</td>\n",
       "      <td>16,011 European ancestry women, 17,912 Europea...</td>\n",
       "      <td>...</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.469</td>\n",
       "      <td>7.000000e-07</td>\n",
       "      <td>6.154902</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0192</td>\n",
       "      <td>[0.012-0.027] kg/m2 increase</td>\n",
       "      <td>Affymetrix, Illumina, Perlegen [up to 2800000]...</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2017-08-10</td>\n",
       "      <td>28443625</td>\n",
       "      <td>Justice AE</td>\n",
       "      <td>2017-04-26</td>\n",
       "      <td>Nat Commun</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/28443625</td>\n",
       "      <td>Genome-wide meta-analysis of 241,258 adults ac...</td>\n",
       "      <td>BMI in non-smokers</td>\n",
       "      <td>77,279 European ancestry women, 47,280 Europea...</td>\n",
       "      <td>16,011 European ancestry women, 17,912 Europea...</td>\n",
       "      <td>...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.2894</td>\n",
       "      <td>7.000000e-09</td>\n",
       "      <td>8.154902</td>\n",
       "      <td>(men)</td>\n",
       "      <td>0.0341</td>\n",
       "      <td>[0.023-0.046] kg/m2 decrease</td>\n",
       "      <td>Affymetrix, Illumina, Perlegen [up to 2800000]...</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2017-08-10</td>\n",
       "      <td>28443625</td>\n",
       "      <td>Justice AE</td>\n",
       "      <td>2017-04-26</td>\n",
       "      <td>Nat Commun</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/28443625</td>\n",
       "      <td>Genome-wide meta-analysis of 241,258 adults ac...</td>\n",
       "      <td>BMI in non-smokers</td>\n",
       "      <td>77,279 European ancestry women, 47,280 Europea...</td>\n",
       "      <td>16,011 European ancestry women, 17,912 Europea...</td>\n",
       "      <td>...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.2894</td>\n",
       "      <td>7.000000e-10</td>\n",
       "      <td>9.154902</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0264</td>\n",
       "      <td>[0.018-0.035] kg/m2 decrease</td>\n",
       "      <td>Affymetrix, Illumina, Perlegen [up to 2800000]...</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2017-08-10</td>\n",
       "      <td>28443625</td>\n",
       "      <td>Justice AE</td>\n",
       "      <td>2017-04-26</td>\n",
       "      <td>Nat Commun</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/28443625</td>\n",
       "      <td>Genome-wide meta-analysis of 241,258 adults ac...</td>\n",
       "      <td>BMI in non-smokers</td>\n",
       "      <td>77,279 European ancestry women, 47,280 Europea...</td>\n",
       "      <td>16,011 European ancestry women, 17,912 Europea...</td>\n",
       "      <td>...</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.8269</td>\n",
       "      <td>3.000000e-09</td>\n",
       "      <td>8.522879</td>\n",
       "      <td>(women)</td>\n",
       "      <td>0.0376</td>\n",
       "      <td>[0.025-0.05] kg/m2 decrease</td>\n",
       "      <td>Affymetrix, Illumina, Perlegen [up to 2800000]...</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  DATE ADDED TO CATALOG  PUBMEDID FIRST AUTHOR        DATE     JOURNAL  \\\n",
       "0            2017-08-10  28443625   Justice AE  2017-04-26  Nat Commun   \n",
       "1            2017-08-10  28443625   Justice AE  2017-04-26  Nat Commun   \n",
       "2            2017-08-10  28443625   Justice AE  2017-04-26  Nat Commun   \n",
       "3            2017-08-10  28443625   Justice AE  2017-04-26  Nat Commun   \n",
       "4            2017-08-10  28443625   Justice AE  2017-04-26  Nat Commun   \n",
       "\n",
       "                                   LINK  \\\n",
       "0  www.ncbi.nlm.nih.gov/pubmed/28443625   \n",
       "1  www.ncbi.nlm.nih.gov/pubmed/28443625   \n",
       "2  www.ncbi.nlm.nih.gov/pubmed/28443625   \n",
       "3  www.ncbi.nlm.nih.gov/pubmed/28443625   \n",
       "4  www.ncbi.nlm.nih.gov/pubmed/28443625   \n",
       "\n",
       "                                               STUDY       DISEASE/TRAIT  \\\n",
       "0  Genome-wide meta-analysis of 241,258 adults ac...  BMI in non-smokers   \n",
       "1  Genome-wide meta-analysis of 241,258 adults ac...  BMI in non-smokers   \n",
       "2  Genome-wide meta-analysis of 241,258 adults ac...  BMI in non-smokers   \n",
       "3  Genome-wide meta-analysis of 241,258 adults ac...  BMI in non-smokers   \n",
       "4  Genome-wide meta-analysis of 241,258 adults ac...  BMI in non-smokers   \n",
       "\n",
       "                                 INITIAL SAMPLE SIZE  \\\n",
       "0  77,279 European ancestry women, 47,280 Europea...   \n",
       "1  77,279 European ancestry women, 47,280 Europea...   \n",
       "2  77,279 European ancestry women, 47,280 Europea...   \n",
       "3  77,279 European ancestry women, 47,280 Europea...   \n",
       "4  77,279 European ancestry women, 47,280 Europea...   \n",
       "\n",
       "                             REPLICATION SAMPLE SIZE  ...             CONTEXT  \\\n",
       "0  16,011 European ancestry women, 17,912 Europea...  ...  intergenic_variant   \n",
       "1  16,011 European ancestry women, 17,912 Europea...  ...  intergenic_variant   \n",
       "2  16,011 European ancestry women, 17,912 Europea...  ...      intron_variant   \n",
       "3  16,011 European ancestry women, 17,912 Europea...  ...      intron_variant   \n",
       "4  16,011 European ancestry women, 17,912 Europea...  ...  intergenic_variant   \n",
       "\n",
       "  INTERGENIC RISK ALLELE FREQUENCY       P-VALUE PVALUE_MLOG P-VALUE (TEXT)  \\\n",
       "0        1.0                 0.847  5.000000e-08    7.301030            NaN   \n",
       "1        1.0                 0.469  7.000000e-07    6.154902            NaN   \n",
       "2        0.0                0.2894  7.000000e-09    8.154902          (men)   \n",
       "3        0.0                0.2894  7.000000e-10    9.154902            NaN   \n",
       "4        1.0                0.8269  3.000000e-09    8.522879        (women)   \n",
       "\n",
       "  OR or BETA                 95% CI (TEXT)  \\\n",
       "0     0.0230  [0.015-0.031] kg/m2 increase   \n",
       "1     0.0192  [0.012-0.027] kg/m2 increase   \n",
       "2     0.0341  [0.023-0.046] kg/m2 decrease   \n",
       "3     0.0264  [0.018-0.035] kg/m2 decrease   \n",
       "4     0.0376   [0.025-0.05] kg/m2 decrease   \n",
       "\n",
       "                          PLATFORM [SNPS PASSING QC]  CNV  \n",
       "0  Affymetrix, Illumina, Perlegen [up to 2800000]...    N  \n",
       "1  Affymetrix, Illumina, Perlegen [up to 2800000]...    N  \n",
       "2  Affymetrix, Illumina, Perlegen [up to 2800000]...    N  \n",
       "3  Affymetrix, Illumina, Perlegen [up to 2800000]...    N  \n",
       "4  Affymetrix, Illumina, Perlegen [up to 2800000]...    N  \n",
       "\n",
       "[5 rows x 34 columns]"
      ]
     },
     "execution_count": 220,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(gwas_raw.shape)\n",
    "gwas_raw.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Drop unnecessary columns. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['DATE ADDED TO CATALOG', 'PUBMEDID', 'FIRST AUTHOR', 'DATE', 'JOURNAL',\n",
       "       'LINK', 'STUDY', 'DISEASE/TRAIT', 'INITIAL SAMPLE SIZE',\n",
       "       'REPLICATION SAMPLE SIZE', 'REGION', 'CHR_ID', 'CHR_POS',\n",
       "       'REPORTED GENE(S)', 'MAPPED_GENE', 'UPSTREAM_GENE_ID',\n",
       "       'DOWNSTREAM_GENE_ID', 'SNP_GENE_IDS', 'UPSTREAM_GENE_DISTANCE',\n",
       "       'DOWNSTREAM_GENE_DISTANCE', 'STRONGEST SNP-RISK ALLELE', 'SNPS',\n",
       "       'MERGED', 'SNP_ID_CURRENT', 'CONTEXT', 'INTERGENIC',\n",
       "       'RISK ALLELE FREQUENCY', 'P-VALUE', 'PVALUE_MLOG', 'P-VALUE (TEXT)',\n",
       "       'OR or BETA', '95% CI (TEXT)', 'PLATFORM [SNPS PASSING QC]', 'CNV'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 221,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_raw.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>MAPPED_GENE</th>\n",
       "      <th>SNPS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BMI in non-smokers</td>\n",
       "      <td>SRRM1P2 - LINC00971</td>\n",
       "      <td>rs6794880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BMI in non-smokers</td>\n",
       "      <td>STARP1 - HNRNPA3P5</td>\n",
       "      <td>rs9540493</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BMI in non-smokers</td>\n",
       "      <td>TCF7L2</td>\n",
       "      <td>rs7903146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BMI in non-smokers</td>\n",
       "      <td>TCF7L2</td>\n",
       "      <td>rs7903146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BMI in non-smokers</td>\n",
       "      <td>RPS17P5 - FTH1P5</td>\n",
       "      <td>rs2207139</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        DISEASE/TRAIT          MAPPED_GENE       SNPS\n",
       "0  BMI in non-smokers  SRRM1P2 - LINC00971  rs6794880\n",
       "1  BMI in non-smokers   STARP1 - HNRNPA3P5  rs9540493\n",
       "2  BMI in non-smokers               TCF7L2  rs7903146\n",
       "3  BMI in non-smokers               TCF7L2  rs7903146\n",
       "4  BMI in non-smokers     RPS17P5 - FTH1P5  rs2207139"
      ]
     },
     "execution_count": 222,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drop_cols = ['DATE ADDED TO CATALOG', 'PUBMEDID', 'FIRST AUTHOR', 'DATE', 'JOURNAL',\n",
    "            'LINK', 'STUDY', 'INITIAL SAMPLE SIZE', 'REPLICATION SAMPLE SIZE', 'REGION',\n",
    "            'CHR_ID', 'CHR_POS', 'REPORTED GENE(S)', 'UPSTREAM_GENE_ID', 'DOWNSTREAM_GENE_ID',\n",
    "            'SNP_GENE_IDS', 'UPSTREAM_GENE_DISTANCE', 'DOWNSTREAM_GENE_DISTANCE', 'STRONGEST SNP-RISK ALLELE',\n",
    "            'MERGED', 'SNP_ID_CURRENT', 'CONTEXT', 'INTERGENIC', 'RISK ALLELE FREQUENCY', 'P-VALUE',\n",
    "            'PVALUE_MLOG', 'P-VALUE (TEXT)', 'OR or BETA', '95% CI (TEXT)', 'PLATFORM [SNPS PASSING QC]', 'CNV']\n",
    "gwas_raw = gwas_raw.drop(columns = drop_cols)\n",
    "gwas_raw.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Only interested in cancer related rows right now, filter on cancer related conditions. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(15456, 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>MAPPED_GENE</th>\n",
       "      <th>SNPS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>149</th>\n",
       "      <td>Survival in pancreatic cancer</td>\n",
       "      <td>PAIP2B</td>\n",
       "      <td>rs113988120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>150</th>\n",
       "      <td>Survival in pancreatic cancer</td>\n",
       "      <td>B4GALT4, B4GALT4-AS1</td>\n",
       "      <td>rs4568126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>151</th>\n",
       "      <td>Survival in pancreatic cancer</td>\n",
       "      <td>RPL21P119 - LINC02177</td>\n",
       "      <td>rs4780973</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>152</th>\n",
       "      <td>Survival in pancreatic cancer</td>\n",
       "      <td>LINC00376 - LINC00395</td>\n",
       "      <td>rs1000589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>Survival in pancreatic cancer</td>\n",
       "      <td>LINC01163 - LINC02667</td>\n",
       "      <td>rs10734079</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     DISEASE/TRAIT            MAPPED_GENE         SNPS\n",
       "149  Survival in pancreatic cancer                 PAIP2B  rs113988120\n",
       "150  Survival in pancreatic cancer   B4GALT4, B4GALT4-AS1    rs4568126\n",
       "151  Survival in pancreatic cancer  RPL21P119 - LINC02177    rs4780973\n",
       "152  Survival in pancreatic cancer  LINC00376 - LINC00395    rs1000589\n",
       "153  Survival in pancreatic cancer  LINC01163 - LINC02667   rs10734079"
      ]
     },
     "execution_count": 223,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# cancer related words\n",
    "cancer_related = ['cancer', 'carcinoma', 'leukemia', 'tumor', 'malignancy', 'glioblastoma',\n",
    "                'melanoma', 'lymphoma', 'sarcoma']\n",
    "\n",
    "# filter on only cancer related rows\n",
    "gwas_raw = gwas_raw[gwas_raw['DISEASE/TRAIT'].str.contains('|'.join(cancer_related), case = False, na = False)]\n",
    "print(gwas_raw.shape)\n",
    "gwas_raw.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Drop range, list, and null mapped gene rows. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 224,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# drop rows where MAPPED_GENE is NA\n",
    "gwas_raw = gwas_raw.dropna(subset = ['MAPPED_GENE'])\n",
    "gwas_raw['MAPPED_GENE'].isna().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(9354, 3)"
      ]
     },
     "execution_count": 225,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# drop rows where MAPPED_GENE is a range or intersection \n",
    "gwas_raw = gwas_raw[~gwas_raw['MAPPED_GENE'].str.contains(' - ') & ~gwas_raw['MAPPED_GENE'].str.contains(' x ')]\n",
    "gwas_raw.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(7577, 3)"
      ]
     },
     "execution_count": 226,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# drop rows where MAPPED_GENE is a list\n",
    "gwas_raw = gwas_raw[~gwas_raw['MAPPED_GENE'].str.contains(',') & ~gwas_raw['MAPPED_GENE'].str.contains(';')]\n",
    "gwas_raw.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>MAPPED_GENE</th>\n",
       "      <th>SNPS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>354963</th>\n",
       "      <td>Gastric cancer</td>\n",
       "      <td>HLA-C</td>\n",
       "      <td>rs1050437</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46342</th>\n",
       "      <td>Barrett's esophagus  or Esophageal adenocarcinoma</td>\n",
       "      <td>LINC00208</td>\n",
       "      <td>rs10108511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421052</th>\n",
       "      <td>Tumor necrosis factor receptor superfamily mem...</td>\n",
       "      <td>PLAUR</td>\n",
       "      <td>rs4760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57044</th>\n",
       "      <td>Squamous cell carcinoma</td>\n",
       "      <td>BRCA2</td>\n",
       "      <td>rs1460816</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>278471</th>\n",
       "      <td>Prostate cancer</td>\n",
       "      <td>CHEK2</td>\n",
       "      <td>rs17886163</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63817</th>\n",
       "      <td>Alanine aminotransferase (ALT) levels after re...</td>\n",
       "      <td>PNPLA3</td>\n",
       "      <td>rs738409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>415522</th>\n",
       "      <td>Colorectal cancer</td>\n",
       "      <td>ARHGAP42</td>\n",
       "      <td>rs55864876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>330715</th>\n",
       "      <td>Pancreatic cancer</td>\n",
       "      <td>ANKRD27</td>\n",
       "      <td>rs150101296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>280530</th>\n",
       "      <td>Prostate cancer</td>\n",
       "      <td>ANO7</td>\n",
       "      <td>rs76832527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>193863</th>\n",
       "      <td>Breast cancer</td>\n",
       "      <td>SLC25A21</td>\n",
       "      <td>rs8003014</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            DISEASE/TRAIT MAPPED_GENE  \\\n",
       "354963                                     Gastric cancer       HLA-C   \n",
       "46342   Barrett's esophagus  or Esophageal adenocarcinoma   LINC00208   \n",
       "421052  Tumor necrosis factor receptor superfamily mem...       PLAUR   \n",
       "57044                             Squamous cell carcinoma       BRCA2   \n",
       "278471                                    Prostate cancer       CHEK2   \n",
       "63817   Alanine aminotransferase (ALT) levels after re...      PNPLA3   \n",
       "415522                                  Colorectal cancer    ARHGAP42   \n",
       "330715                                  Pancreatic cancer     ANKRD27   \n",
       "280530                                    Prostate cancer        ANO7   \n",
       "193863                                      Breast cancer    SLC25A21   \n",
       "\n",
       "               SNPS  \n",
       "354963    rs1050437  \n",
       "46342    rs10108511  \n",
       "421052       rs4760  \n",
       "57044     rs1460816  \n",
       "278471   rs17886163  \n",
       "63817      rs738409  \n",
       "415522   rs55864876  \n",
       "330715  rs150101296  \n",
       "280530   rs76832527  \n",
       "193863    rs8003014  "
      ]
     },
     "execution_count": 227,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_raw.sample(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save to tsv \n",
    "gwas_raw.to_csv('../home/data/processed_data/GlyGen/gwas/gwas_processed.tsv', sep = '\\t', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# isolate conditions\n",
    "conditions = gwas_raw['DISEASE/TRAIT']\n",
    "# consolidate unique conditions\n",
    "conditions = conditions.drop_duplicates()\n",
    "# conditions.to_csv('../home/data/processed_data/gwas_conditions.tsv', sep = '\\t', index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Secondary Wrangling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(7577, 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>MAPPED_GENE</th>\n",
       "      <th>SNPS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Survival in pancreatic cancer</td>\n",
       "      <td>PAIP2B</td>\n",
       "      <td>rs113988120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Survival in pancreatic cancer</td>\n",
       "      <td>ACOT11</td>\n",
       "      <td>rs10736390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Survival in pancreatic cancer</td>\n",
       "      <td>BDNF-AS</td>\n",
       "      <td>rs10767646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Survival in pancreatic cancer</td>\n",
       "      <td>WHRN</td>\n",
       "      <td>rs10817611</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Survival in pancreatic cancer</td>\n",
       "      <td>LIN7C</td>\n",
       "      <td>rs10835188</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   DISEASE/TRAIT MAPPED_GENE         SNPS\n",
       "0  Survival in pancreatic cancer      PAIP2B  rs113988120\n",
       "1  Survival in pancreatic cancer      ACOT11   rs10736390\n",
       "2  Survival in pancreatic cancer     BDNF-AS   rs10767646\n",
       "3  Survival in pancreatic cancer        WHRN   rs10817611\n",
       "4  Survival in pancreatic cancer       LIN7C   rs10835188"
      ]
     },
     "execution_count": 229,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_processed = pd.read_csv('../home/data/processed_data/GlyGen/gwas/gwas_processed.tsv', sep='\\t')\n",
    "print(gwas_processed.shape)\n",
    "gwas_processed.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Clean conditions. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "metadata": {},
   "outputs": [],
   "source": [
    "gwas_conditions = pd.read_csv('../home/data/processed_data/GlyGen/gwas/gwas_conditions.tsv', sep = '\\t')\n",
    "gwas_drop_conditions = pd.read_csv('../home/data/processed_data/GlyGen/gwas/gwas_drop_conditions.tsv', sep = '\\t')\n",
    "gwas_explode_conditions = pd.read_csv('../home/data/processed_data/GlyGen/gwas/gwas_explode_conditions.tsv', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {},
   "outputs": [],
   "source": [
    "drop_conditions = set(gwas_drop_conditions['condition'])\n",
    "gwas_processed = gwas_processed[~gwas_processed['DISEASE/TRAIT'].isin(drop_conditions)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Isolate single and list condtions. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(198, 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>MAPPED_GENE</th>\n",
       "      <th>SNPS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>Breast cancer, ovarian cancer or prostate canc...</td>\n",
       "      <td>ZBTB7B</td>\n",
       "      <td>rs77548871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>Breast cancer, ovarian cancer or prostate canc...</td>\n",
       "      <td>VAMP8</td>\n",
       "      <td>rs13007211</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Breast cancer, ovarian cancer or prostate canc...</td>\n",
       "      <td>JAZF1</td>\n",
       "      <td>rs739704</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Breast cancer, ovarian cancer or prostate canc...</td>\n",
       "      <td>ARHGEF5</td>\n",
       "      <td>rs720475</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>Breast cancer, ovarian cancer or prostate canc...</td>\n",
       "      <td>PCAT1</td>\n",
       "      <td>rs17762342</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        DISEASE/TRAIT MAPPED_GENE        SNPS\n",
       "38  Breast cancer, ovarian cancer or prostate canc...      ZBTB7B  rs77548871\n",
       "39  Breast cancer, ovarian cancer or prostate canc...       VAMP8  rs13007211\n",
       "40  Breast cancer, ovarian cancer or prostate canc...       JAZF1    rs739704\n",
       "41  Breast cancer, ovarian cancer or prostate canc...     ARHGEF5    rs720475\n",
       "42  Breast cancer, ovarian cancer or prostate canc...       PCAT1  rs17762342"
      ]
     },
     "execution_count": 232,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_explode_condition_rows = gwas_processed[gwas_processed['DISEASE/TRAIT'].isin(set(gwas_explode_conditions['raw_disease']))]\n",
    "print(gwas_explode_condition_rows.shape)\n",
    "gwas_explode_condition_rows.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(7045, 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>MAPPED_GENE</th>\n",
       "      <th>SNPS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Survival in pancreatic cancer</td>\n",
       "      <td>PAIP2B</td>\n",
       "      <td>rs113988120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Survival in pancreatic cancer</td>\n",
       "      <td>ACOT11</td>\n",
       "      <td>rs10736390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Survival in pancreatic cancer</td>\n",
       "      <td>BDNF-AS</td>\n",
       "      <td>rs10767646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Survival in pancreatic cancer</td>\n",
       "      <td>WHRN</td>\n",
       "      <td>rs10817611</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Survival in pancreatic cancer</td>\n",
       "      <td>LIN7C</td>\n",
       "      <td>rs10835188</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   DISEASE/TRAIT MAPPED_GENE         SNPS\n",
       "0  Survival in pancreatic cancer      PAIP2B  rs113988120\n",
       "1  Survival in pancreatic cancer      ACOT11   rs10736390\n",
       "2  Survival in pancreatic cancer     BDNF-AS   rs10767646\n",
       "3  Survival in pancreatic cancer        WHRN   rs10817611\n",
       "4  Survival in pancreatic cancer       LIN7C   rs10835188"
      ]
     },
     "execution_count": 233,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_single_condition_rows = gwas_processed[gwas_processed['DISEASE/TRAIT'].isin(set(gwas_conditions['DISEASE/TRAIT']))]\n",
    "print(gwas_single_condition_rows.shape) \n",
    "gwas_single_condition_rows.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "metadata": {},
   "outputs": [],
   "source": [
    "# sanity check\n",
    "assert(gwas_processed.shape[0] == gwas_explode_condition_rows.shape[0] + gwas_single_condition_rows.shape[0])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Map single condition rows."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>MAPPED_GENE</th>\n",
       "      <th>SNPS</th>\n",
       "      <th>condition</th>\n",
       "      <th>doid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4377</th>\n",
       "      <td>Colorectal cancer</td>\n",
       "      <td>OSBP2</td>\n",
       "      <td>rs926340</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>:9256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5833</th>\n",
       "      <td>Uterine leiomyoma or breast cancer (pleiotropy)</td>\n",
       "      <td>ATAD5</td>\n",
       "      <td>rs117662433</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>:1612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2472</th>\n",
       "      <td>Squamous cell lung carcinoma</td>\n",
       "      <td>CLPTM1L</td>\n",
       "      <td>rs467095</td>\n",
       "      <td>lung squamous cell carcinoma</td>\n",
       "      <td>:3907</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1895</th>\n",
       "      <td>Nevus count or cutaneous melanoma</td>\n",
       "      <td>CDH3</td>\n",
       "      <td>rs8046299</td>\n",
       "      <td>skin melanoma</td>\n",
       "      <td>:8923</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>Prostate cancer</td>\n",
       "      <td>RASSF3</td>\n",
       "      <td>rs7968403</td>\n",
       "      <td>prostate cancer</td>\n",
       "      <td>:10283</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        DISEASE/TRAIT MAPPED_GENE  \\\n",
       "4377                                Colorectal cancer       OSBP2   \n",
       "5833  Uterine leiomyoma or breast cancer (pleiotropy)       ATAD5   \n",
       "2472                     Squamous cell lung carcinoma     CLPTM1L   \n",
       "1895                Nevus count or cutaneous melanoma        CDH3   \n",
       "296                                   Prostate cancer      RASSF3   \n",
       "\n",
       "             SNPS                     condition    doid  \n",
       "4377     rs926340             colorectal cancer   :9256  \n",
       "5833  rs117662433                 breast cancer   :1612  \n",
       "2472     rs467095  lung squamous cell carcinoma   :3907  \n",
       "1895    rs8046299                 skin melanoma   :8923  \n",
       "296     rs7968403               prostate cancer  :10283  "
      ]
     },
     "execution_count": 235,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# merge single condition rows with condition list\n",
    "gwas_single_condition_rows = gwas_single_condition_rows.merge(gwas_conditions, how = 'left', on = 'DISEASE/TRAIT')\n",
    "gwas_single_condition_rows.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DISEASE/TRAIT    0\n",
       "MAPPED_GENE      0\n",
       "SNPS             0\n",
       "condition        0\n",
       "doid             0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 236,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_single_condition_rows.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 237,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MAPPED_GENE</th>\n",
       "      <th>SNPS</th>\n",
       "      <th>condition</th>\n",
       "      <th>doid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1551</th>\n",
       "      <td>HELQ</td>\n",
       "      <td>rs1494961</td>\n",
       "      <td>oral cavity cancer</td>\n",
       "      <td>:8618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5368</th>\n",
       "      <td>CNIH3</td>\n",
       "      <td>rs72761829</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>:1612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6019</th>\n",
       "      <td>LINC02540</td>\n",
       "      <td>rs116995626</td>\n",
       "      <td>colorectal cancer</td>\n",
       "      <td>:9256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6419</th>\n",
       "      <td>MAST2</td>\n",
       "      <td>rs4420029</td>\n",
       "      <td>prostate cancer</td>\n",
       "      <td>:10283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4274</th>\n",
       "      <td>NAPEPLD</td>\n",
       "      <td>rs56196003</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>:1612</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     MAPPED_GENE         SNPS           condition    doid\n",
       "1551        HELQ    rs1494961  oral cavity cancer   :8618\n",
       "5368       CNIH3   rs72761829       breast cancer   :1612\n",
       "6019   LINC02540  rs116995626   colorectal cancer   :9256\n",
       "6419       MAST2    rs4420029     prostate cancer  :10283\n",
       "4274     NAPEPLD   rs56196003       breast cancer   :1612"
      ]
     },
     "execution_count": 237,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_single_condition_rows = gwas_single_condition_rows.drop('DISEASE/TRAIT', axis = 1)\n",
    "gwas_single_condition_rows.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>MAPPED_GENE</th>\n",
       "      <th>SNPS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3790</th>\n",
       "      <td>Breast cancer or ovarian cancer (pleiotropy)</td>\n",
       "      <td>ABHD8</td>\n",
       "      <td>rs4808616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3796</th>\n",
       "      <td>Breast cancer or ovarian cancer (pleiotropy)</td>\n",
       "      <td>ADAM29</td>\n",
       "      <td>rs6826366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5520</th>\n",
       "      <td>Endometriosis or endometrial cancer (pleiotropy)</td>\n",
       "      <td>PTPRD</td>\n",
       "      <td>rs2475335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3797</th>\n",
       "      <td>Breast cancer or ovarian cancer (pleiotropy)</td>\n",
       "      <td>TERT</td>\n",
       "      <td>rs10069690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3737</th>\n",
       "      <td>Breast cancer or lung cancer (pleiotropy)</td>\n",
       "      <td>HSPA4</td>\n",
       "      <td>rs13718</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         DISEASE/TRAIT MAPPED_GENE        SNPS\n",
       "3790      Breast cancer or ovarian cancer (pleiotropy)       ABHD8   rs4808616\n",
       "3796      Breast cancer or ovarian cancer (pleiotropy)      ADAM29   rs6826366\n",
       "5520  Endometriosis or endometrial cancer (pleiotropy)       PTPRD   rs2475335\n",
       "3797      Breast cancer or ovarian cancer (pleiotropy)        TERT  rs10069690\n",
       "3737         Breast cancer or lung cancer (pleiotropy)       HSPA4     rs13718"
      ]
     },
     "execution_count": 238,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_single_condition_rows['doid'] = gwas_single_condition_rows['doid'].str.split(':').str[1]\n",
    "gwas_explode_condition_rows.sample(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Map multiple condition rows."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>MAPPED_GENE</th>\n",
       "      <th>SNPS</th>\n",
       "      <th>raw_disease</th>\n",
       "      <th>list_diseases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>Malignant lymphoma (Non-Hodgkin lymphoma, mult...</td>\n",
       "      <td>QPCT</td>\n",
       "      <td>rs3770745</td>\n",
       "      <td>Malignant lymphoma (Non-Hodgkin lymphoma, mult...</td>\n",
       "      <td>non-Hodgkin lymphoma:0060060,multiple myeloma:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Colorectal or endometrial cancer</td>\n",
       "      <td>LSAMP</td>\n",
       "      <td>rs4378954</td>\n",
       "      <td>Colorectal or endometrial cancer</td>\n",
       "      <td>colorectal cancer:9256,endometrial cancer:1380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>Breast cancer or ovarian cancer (pleiotropy)</td>\n",
       "      <td>NRIP1</td>\n",
       "      <td>rs2822991</td>\n",
       "      <td>Breast cancer or ovarian cancer (pleiotropy)</td>\n",
       "      <td>breast cancer:1612,ovarian cancer:2394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>Breast cancer or lung cancer (pleiotropy)</td>\n",
       "      <td>ABHD8</td>\n",
       "      <td>rs4808616</td>\n",
       "      <td>Breast cancer or lung cancer (pleiotropy)</td>\n",
       "      <td>breast cancer:1612,lung cancer:1324</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>Barrett's esophagus or esophageal adenocarcino...</td>\n",
       "      <td>DPYSL2</td>\n",
       "      <td>rs17321041</td>\n",
       "      <td>Barrett's esophagus or esophageal adenocarcino...</td>\n",
       "      <td>Barrett's esophagus:9206,esophagus adenocarcin...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         DISEASE/TRAIT MAPPED_GENE  \\\n",
       "174  Malignant lymphoma (Non-Hodgkin lymphoma, mult...        QPCT   \n",
       "23                    Colorectal or endometrial cancer       LSAMP   \n",
       "91        Breast cancer or ovarian cancer (pleiotropy)       NRIP1   \n",
       "77           Breast cancer or lung cancer (pleiotropy)       ABHD8   \n",
       "127  Barrett's esophagus or esophageal adenocarcino...      DPYSL2   \n",
       "\n",
       "           SNPS                                        raw_disease  \\\n",
       "174   rs3770745  Malignant lymphoma (Non-Hodgkin lymphoma, mult...   \n",
       "23    rs4378954                   Colorectal or endometrial cancer   \n",
       "91    rs2822991       Breast cancer or ovarian cancer (pleiotropy)   \n",
       "77    rs4808616          Breast cancer or lung cancer (pleiotropy)   \n",
       "127  rs17321041  Barrett's esophagus or esophageal adenocarcino...   \n",
       "\n",
       "                                         list_diseases  \n",
       "174  non-Hodgkin lymphoma:0060060,multiple myeloma:...  \n",
       "23      colorectal cancer:9256,endometrial cancer:1380  \n",
       "91              breast cancer:1612,ovarian cancer:2394  \n",
       "77                 breast cancer:1612,lung cancer:1324  \n",
       "127  Barrett's esophagus:9206,esophagus adenocarcin...  "
      ]
     },
     "execution_count": 239,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# merge explode condition rows with explode condition list\n",
    "gwas_explode_condition_rows = gwas_explode_condition_rows.merge(gwas_explode_conditions, how = 'left', left_on = 'DISEASE/TRAIT', right_on = 'raw_disease')\n",
    "gwas_explode_condition_rows.sample(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DISEASE/TRAIT    0\n",
       "MAPPED_GENE      0\n",
       "SNPS             0\n",
       "raw_disease      0\n",
       "list_diseases    0\n",
       "dtype: int64"
      ]
     },
     "execution_count": 240,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_explode_condition_rows.isnull().sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MAPPED_GENE</th>\n",
       "      <th>SNPS</th>\n",
       "      <th>list_diseases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZBTB7B</td>\n",
       "      <td>rs77548871</td>\n",
       "      <td>breast cancer:1612,ovarian cancer:2394,prostat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VAMP8</td>\n",
       "      <td>rs13007211</td>\n",
       "      <td>breast cancer:1612,ovarian cancer:2394,prostat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>JAZF1</td>\n",
       "      <td>rs739704</td>\n",
       "      <td>breast cancer:1612,ovarian cancer:2394,prostat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARHGEF5</td>\n",
       "      <td>rs720475</td>\n",
       "      <td>breast cancer:1612,ovarian cancer:2394,prostat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PCAT1</td>\n",
       "      <td>rs17762342</td>\n",
       "      <td>breast cancer:1612,ovarian cancer:2394,prostat...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  MAPPED_GENE        SNPS                                      list_diseases\n",
       "0      ZBTB7B  rs77548871  breast cancer:1612,ovarian cancer:2394,prostat...\n",
       "1       VAMP8  rs13007211  breast cancer:1612,ovarian cancer:2394,prostat...\n",
       "2       JAZF1    rs739704  breast cancer:1612,ovarian cancer:2394,prostat...\n",
       "3     ARHGEF5    rs720475  breast cancer:1612,ovarian cancer:2394,prostat...\n",
       "4       PCAT1  rs17762342  breast cancer:1612,ovarian cancer:2394,prostat..."
      ]
     },
     "execution_count": 241,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_explode_condition_rows = gwas_explode_condition_rows.drop(['DISEASE/TRAIT', 'raw_disease'], axis = 1)\n",
    "gwas_explode_condition_rows.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MAPPED_GENE</th>\n",
       "      <th>SNPS</th>\n",
       "      <th>list_diseases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZBTB7B</td>\n",
       "      <td>rs77548871</td>\n",
       "      <td>[breast cancer:1612, ovarian cancer:2394, pros...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VAMP8</td>\n",
       "      <td>rs13007211</td>\n",
       "      <td>[breast cancer:1612, ovarian cancer:2394, pros...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>JAZF1</td>\n",
       "      <td>rs739704</td>\n",
       "      <td>[breast cancer:1612, ovarian cancer:2394, pros...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ARHGEF5</td>\n",
       "      <td>rs720475</td>\n",
       "      <td>[breast cancer:1612, ovarian cancer:2394, pros...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PCAT1</td>\n",
       "      <td>rs17762342</td>\n",
       "      <td>[breast cancer:1612, ovarian cancer:2394, pros...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  MAPPED_GENE        SNPS                                      list_diseases\n",
       "0      ZBTB7B  rs77548871  [breast cancer:1612, ovarian cancer:2394, pros...\n",
       "1       VAMP8  rs13007211  [breast cancer:1612, ovarian cancer:2394, pros...\n",
       "2       JAZF1    rs739704  [breast cancer:1612, ovarian cancer:2394, pros...\n",
       "3     ARHGEF5    rs720475  [breast cancer:1612, ovarian cancer:2394, pros...\n",
       "4       PCAT1  rs17762342  [breast cancer:1612, ovarian cancer:2394, pros..."
      ]
     },
     "execution_count": 242,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_explode_condition_rows['list_diseases'] = gwas_explode_condition_rows['list_diseases'].str.split(',')\n",
    "gwas_explode_condition_rows.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(430, 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MAPPED_GENE</th>\n",
       "      <th>SNPS</th>\n",
       "      <th>list_diseases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZBTB7B</td>\n",
       "      <td>rs77548871</td>\n",
       "      <td>breast cancer:1612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZBTB7B</td>\n",
       "      <td>rs77548871</td>\n",
       "      <td>ovarian cancer:2394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZBTB7B</td>\n",
       "      <td>rs77548871</td>\n",
       "      <td>prostate cancer:10283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VAMP8</td>\n",
       "      <td>rs13007211</td>\n",
       "      <td>breast cancer:1612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VAMP8</td>\n",
       "      <td>rs13007211</td>\n",
       "      <td>ovarian cancer:2394</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  MAPPED_GENE        SNPS          list_diseases\n",
       "0      ZBTB7B  rs77548871     breast cancer:1612\n",
       "0      ZBTB7B  rs77548871    ovarian cancer:2394\n",
       "0      ZBTB7B  rs77548871  prostate cancer:10283\n",
       "1       VAMP8  rs13007211     breast cancer:1612\n",
       "1       VAMP8  rs13007211    ovarian cancer:2394"
      ]
     },
     "execution_count": 243,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# explode condition list\n",
    "gwas_explode_condition_rows = gwas_explode_condition_rows.explode('list_diseases')\n",
    "print(gwas_explode_condition_rows.shape)\n",
    "gwas_explode_condition_rows.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Split separate DOID and condition names."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(430, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MAPPED_GENE</th>\n",
       "      <th>SNPS</th>\n",
       "      <th>condition</th>\n",
       "      <th>doid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZBTB7B</td>\n",
       "      <td>rs77548871</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>1612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZBTB7B</td>\n",
       "      <td>rs77548871</td>\n",
       "      <td>ovarian cancer</td>\n",
       "      <td>2394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZBTB7B</td>\n",
       "      <td>rs77548871</td>\n",
       "      <td>prostate cancer</td>\n",
       "      <td>10283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VAMP8</td>\n",
       "      <td>rs13007211</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>1612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>VAMP8</td>\n",
       "      <td>rs13007211</td>\n",
       "      <td>ovarian cancer</td>\n",
       "      <td>2394</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  MAPPED_GENE        SNPS        condition   doid\n",
       "0      ZBTB7B  rs77548871    breast cancer   1612\n",
       "0      ZBTB7B  rs77548871   ovarian cancer   2394\n",
       "0      ZBTB7B  rs77548871  prostate cancer  10283\n",
       "1       VAMP8  rs13007211    breast cancer   1612\n",
       "1       VAMP8  rs13007211   ovarian cancer   2394"
      ]
     },
     "execution_count": 244,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def split_condition_doid(row):\n",
    "    row['condition'] = row['list_diseases'].split(':')[0]\n",
    "    row['doid'] = row['list_diseases'].split(':')[1]\n",
    "    return row\n",
    "\n",
    "gwas_explode_condition_rows = gwas_explode_condition_rows.apply(split_condition_doid, axis = 1)\n",
    "gwas_explode_condition_rows = gwas_explode_condition_rows.drop('list_diseases', axis = 1)\n",
    "print(gwas_explode_condition_rows.shape)\n",
    "gwas_explode_condition_rows.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Recombine data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MAPPED_GENE</th>\n",
       "      <th>SNPS</th>\n",
       "      <th>condition</th>\n",
       "      <th>doid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4626</th>\n",
       "      <td>LHPP</td>\n",
       "      <td>rs35837782</td>\n",
       "      <td>acute lymphoblastic leukemia</td>\n",
       "      <td>9952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2468</th>\n",
       "      <td>NEB</td>\n",
       "      <td>rs10174077</td>\n",
       "      <td>lung squamous cell carcinoma</td>\n",
       "      <td>3907</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6861</th>\n",
       "      <td>MSMB</td>\n",
       "      <td>rs10993994</td>\n",
       "      <td>prostate cancer</td>\n",
       "      <td>10283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>418</th>\n",
       "      <td>LINGO2</td>\n",
       "      <td>rs2891316</td>\n",
       "      <td>endometrial cancer</td>\n",
       "      <td>1380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>868</th>\n",
       "      <td>MYO9B</td>\n",
       "      <td>rs7249698</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>1612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3845</th>\n",
       "      <td>ATM</td>\n",
       "      <td>rs1801516</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>1909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5825</th>\n",
       "      <td>TERT</td>\n",
       "      <td>rs56345976</td>\n",
       "      <td>breast cancer</td>\n",
       "      <td>1612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3295</th>\n",
       "      <td>LPP</td>\n",
       "      <td>rs1464510</td>\n",
       "      <td>skin cancer</td>\n",
       "      <td>4159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1320</th>\n",
       "      <td>PRPSAP1</td>\n",
       "      <td>rs66459581</td>\n",
       "      <td>ovarian cancer</td>\n",
       "      <td>2394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6934</th>\n",
       "      <td>LINC01169</td>\n",
       "      <td>rs8023793</td>\n",
       "      <td>prostate cancer</td>\n",
       "      <td>10283</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     MAPPED_GENE        SNPS                     condition   doid\n",
       "4626        LHPP  rs35837782  acute lymphoblastic leukemia   9952\n",
       "2468         NEB  rs10174077  lung squamous cell carcinoma   3907\n",
       "6861        MSMB  rs10993994               prostate cancer  10283\n",
       "418       LINGO2   rs2891316            endometrial cancer   1380\n",
       "868        MYO9B   rs7249698                 breast cancer   1612\n",
       "3845         ATM   rs1801516                      melanoma   1909\n",
       "5825        TERT  rs56345976                 breast cancer   1612\n",
       "3295         LPP   rs1464510                   skin cancer   4159\n",
       "1320     PRPSAP1  rs66459581                ovarian cancer   2394\n",
       "6934   LINC01169   rs8023793               prostate cancer  10283"
      ]
     },
     "execution_count": 245,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# combine single and explode condition rows\n",
    "gwas_processed = pd.concat([gwas_single_condition_rows, gwas_explode_condition_rows])\n",
    "gwas_processed.sample(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final Mappings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 260,
   "metadata": {},
   "outputs": [],
   "source": [
    "# gwas_processed.to_csv('../home/data/processed_data/GlyGen/gwas/gwas_final_processed.tsv', sep = '\\t', index = False)\n",
    "gwas_processed = pd.read_csv('../home/data/processed_data/GlyGen/gwas/gwas_final_processed.tsv', sep = '\\t', dtype={'doid': str})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Create the empty TSV file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biomarker_id</th>\n",
       "      <th>biomarker</th>\n",
       "      <th>assessed_biomarker_entity</th>\n",
       "      <th>assessed_biomarker_entity_id</th>\n",
       "      <th>assessed_entity_type</th>\n",
       "      <th>condition</th>\n",
       "      <th>condition_id</th>\n",
       "      <th>exposure_agent</th>\n",
       "      <th>exposure_agent_id</th>\n",
       "      <th>best_biomarker_role</th>\n",
       "      <th>specimen</th>\n",
       "      <th>specimen_id</th>\n",
       "      <th>loinc_code</th>\n",
       "      <th>evidence_source</th>\n",
       "      <th>evidence</th>\n",
       "      <th>tag</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [biomarker_id, biomarker, assessed_biomarker_entity, assessed_biomarker_entity_id, assessed_entity_type, condition, condition_id, exposure_agent, exposure_agent_id, best_biomarker_role, specimen, specimen_id, loinc_code, evidence_source, evidence, tag]\n",
       "Index: []"
      ]
     },
     "execution_count": 261,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame(columns = tcf.TSV_HEADERS)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Final mappings."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(7510, 16)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biomarker_id</th>\n",
       "      <th>biomarker</th>\n",
       "      <th>assessed_biomarker_entity</th>\n",
       "      <th>assessed_biomarker_entity_id</th>\n",
       "      <th>assessed_entity_type</th>\n",
       "      <th>condition</th>\n",
       "      <th>condition_id</th>\n",
       "      <th>exposure_agent</th>\n",
       "      <th>exposure_agent_id</th>\n",
       "      <th>best_biomarker_role</th>\n",
       "      <th>specimen</th>\n",
       "      <th>specimen_id</th>\n",
       "      <th>loinc_code</th>\n",
       "      <th>evidence_source</th>\n",
       "      <th>evidence</th>\n",
       "      <th>tag</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>presence of rs113988120 mutation in PAIP2B</td>\n",
       "      <td>PAIP2B</td>\n",
       "      <td>dbSNP:rs113988120</td>\n",
       "      <td>gene</td>\n",
       "      <td>pancreatic cancer</td>\n",
       "      <td>DOID:1793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GWAS:PAIP2B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>presence of rs10736390 mutation in ACOT11</td>\n",
       "      <td>ACOT11</td>\n",
       "      <td>dbSNP:rs10736390</td>\n",
       "      <td>gene</td>\n",
       "      <td>pancreatic cancer</td>\n",
       "      <td>DOID:1793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GWAS:ACOT11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>presence of rs10767646 mutation in BDNF-AS</td>\n",
       "      <td>BDNF-AS</td>\n",
       "      <td>dbSNP:rs10767646</td>\n",
       "      <td>gene</td>\n",
       "      <td>pancreatic cancer</td>\n",
       "      <td>DOID:1793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GWAS:BDNF-AS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>presence of rs10817611 mutation in WHRN</td>\n",
       "      <td>WHRN</td>\n",
       "      <td>dbSNP:rs10817611</td>\n",
       "      <td>gene</td>\n",
       "      <td>pancreatic cancer</td>\n",
       "      <td>DOID:1793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GWAS:WHRN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>presence of rs10835188 mutation in LIN7C</td>\n",
       "      <td>LIN7C</td>\n",
       "      <td>dbSNP:rs10835188</td>\n",
       "      <td>gene</td>\n",
       "      <td>pancreatic cancer</td>\n",
       "      <td>DOID:1793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GWAS:LIN7C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  biomarker_id                                   biomarker  \\\n",
       "0          NaN  presence of rs113988120 mutation in PAIP2B   \n",
       "1          NaN   presence of rs10736390 mutation in ACOT11   \n",
       "2          NaN  presence of rs10767646 mutation in BDNF-AS   \n",
       "3          NaN     presence of rs10817611 mutation in WHRN   \n",
       "4          NaN    presence of rs10835188 mutation in LIN7C   \n",
       "\n",
       "  assessed_biomarker_entity assessed_biomarker_entity_id assessed_entity_type  \\\n",
       "0                    PAIP2B            dbSNP:rs113988120                 gene   \n",
       "1                    ACOT11             dbSNP:rs10736390                 gene   \n",
       "2                   BDNF-AS             dbSNP:rs10767646                 gene   \n",
       "3                      WHRN             dbSNP:rs10817611                 gene   \n",
       "4                     LIN7C             dbSNP:rs10835188                 gene   \n",
       "\n",
       "           condition condition_id exposure_agent exposure_agent_id  \\\n",
       "0  pancreatic cancer    DOID:1793            NaN               NaN   \n",
       "1  pancreatic cancer    DOID:1793            NaN               NaN   \n",
       "2  pancreatic cancer    DOID:1793            NaN               NaN   \n",
       "3  pancreatic cancer    DOID:1793            NaN               NaN   \n",
       "4  pancreatic cancer    DOID:1793            NaN               NaN   \n",
       "\n",
       "  best_biomarker_role specimen specimen_id loinc_code evidence_source  \\\n",
       "0                risk      NaN         NaN        NaN     GWAS:PAIP2B   \n",
       "1                risk      NaN         NaN        NaN     GWAS:ACOT11   \n",
       "2                risk      NaN         NaN        NaN    GWAS:BDNF-AS   \n",
       "3                risk      NaN         NaN        NaN       GWAS:WHRN   \n",
       "4                risk      NaN         NaN        NaN      GWAS:LIN7C   \n",
       "\n",
       "  evidence                                                tag  \n",
       "0      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "1      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "2      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "3      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "4      NaN  biomarker;assessed_biomarker_entity_id;assesse...  "
      ]
     },
     "execution_count": 262,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['assessed_biomarker_entity'] = gwas_processed['MAPPED_GENE']\n",
    "df['biomarker'] = 'presence of ' + gwas_processed['SNPS'].astype(str) + ' mutation in ' + df['assessed_biomarker_entity']\n",
    "df['assessed_biomarker_entity_id'] = 'dbSNP:' + gwas_processed['SNPS'].astype(str)\n",
    "df['assessed_entity_type'] = 'gene'\n",
    "df['condition'] = gwas_processed['condition'].str.strip()\n",
    "df['condition_id'] = 'DOID:' + gwas_processed['doid'].astype(str).str.strip()\n",
    "df['best_biomarker_role'] = 'risk'\n",
    "df['evidence_source'] = 'GWAS:' + gwas_processed['MAPPED_GENE'].astype(str).str.strip()\n",
    "df['tag'] = 'biomarker;assessed_biomarker_entity_id;assessed_biomarker_entity;specimen;condition'\n",
    "\n",
    "print(df.shape)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 266,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5432, 17)"
      ]
     },
     "execution_count": 266,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# drop rows that are full duplicates\n",
    "df = df.drop_duplicates()\n",
    "df.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Assign temporary IDs."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 264,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tmp_id</th>\n",
       "      <th>biomarker_id</th>\n",
       "      <th>biomarker</th>\n",
       "      <th>assessed_biomarker_entity</th>\n",
       "      <th>assessed_biomarker_entity_id</th>\n",
       "      <th>assessed_entity_type</th>\n",
       "      <th>condition</th>\n",
       "      <th>condition_id</th>\n",
       "      <th>exposure_agent</th>\n",
       "      <th>exposure_agent_id</th>\n",
       "      <th>best_biomarker_role</th>\n",
       "      <th>specimen</th>\n",
       "      <th>specimen_id</th>\n",
       "      <th>loinc_code</th>\n",
       "      <th>evidence_source</th>\n",
       "      <th>evidence</th>\n",
       "      <th>tag</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>533</td>\n",
       "      <td>NaN</td>\n",
       "      <td>presence of rs113988120 mutation in PAIP2B</td>\n",
       "      <td>PAIP2B</td>\n",
       "      <td>dbSNP:rs113988120</td>\n",
       "      <td>gene</td>\n",
       "      <td>pancreatic cancer</td>\n",
       "      <td>DOID:1793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GWAS:PAIP2B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>176</td>\n",
       "      <td>NaN</td>\n",
       "      <td>presence of rs10736390 mutation in ACOT11</td>\n",
       "      <td>ACOT11</td>\n",
       "      <td>dbSNP:rs10736390</td>\n",
       "      <td>gene</td>\n",
       "      <td>pancreatic cancer</td>\n",
       "      <td>DOID:1793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GWAS:ACOT11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>193</td>\n",
       "      <td>NaN</td>\n",
       "      <td>presence of rs10767646 mutation in BDNF-AS</td>\n",
       "      <td>BDNF-AS</td>\n",
       "      <td>dbSNP:rs10767646</td>\n",
       "      <td>gene</td>\n",
       "      <td>pancreatic cancer</td>\n",
       "      <td>DOID:1793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GWAS:BDNF-AS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>218</td>\n",
       "      <td>NaN</td>\n",
       "      <td>presence of rs10817611 mutation in WHRN</td>\n",
       "      <td>WHRN</td>\n",
       "      <td>dbSNP:rs10817611</td>\n",
       "      <td>gene</td>\n",
       "      <td>pancreatic cancer</td>\n",
       "      <td>DOID:1793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GWAS:WHRN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>229</td>\n",
       "      <td>NaN</td>\n",
       "      <td>presence of rs10835188 mutation in LIN7C</td>\n",
       "      <td>LIN7C</td>\n",
       "      <td>dbSNP:rs10835188</td>\n",
       "      <td>gene</td>\n",
       "      <td>pancreatic cancer</td>\n",
       "      <td>DOID:1793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GWAS:LIN7C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   tmp_id biomarker_id                                   biomarker  \\\n",
       "0     533          NaN  presence of rs113988120 mutation in PAIP2B   \n",
       "1     176          NaN   presence of rs10736390 mutation in ACOT11   \n",
       "2     193          NaN  presence of rs10767646 mutation in BDNF-AS   \n",
       "3     218          NaN     presence of rs10817611 mutation in WHRN   \n",
       "4     229          NaN    presence of rs10835188 mutation in LIN7C   \n",
       "\n",
       "  assessed_biomarker_entity assessed_biomarker_entity_id assessed_entity_type  \\\n",
       "0                    PAIP2B            dbSNP:rs113988120                 gene   \n",
       "1                    ACOT11             dbSNP:rs10736390                 gene   \n",
       "2                   BDNF-AS             dbSNP:rs10767646                 gene   \n",
       "3                      WHRN             dbSNP:rs10817611                 gene   \n",
       "4                     LIN7C             dbSNP:rs10835188                 gene   \n",
       "\n",
       "           condition condition_id exposure_agent exposure_agent_id  \\\n",
       "0  pancreatic cancer    DOID:1793            NaN               NaN   \n",
       "1  pancreatic cancer    DOID:1793            NaN               NaN   \n",
       "2  pancreatic cancer    DOID:1793            NaN               NaN   \n",
       "3  pancreatic cancer    DOID:1793            NaN               NaN   \n",
       "4  pancreatic cancer    DOID:1793            NaN               NaN   \n",
       "\n",
       "  best_biomarker_role specimen specimen_id loinc_code evidence_source  \\\n",
       "0                risk      NaN         NaN        NaN     GWAS:PAIP2B   \n",
       "1                risk      NaN         NaN        NaN     GWAS:ACOT11   \n",
       "2                risk      NaN         NaN        NaN    GWAS:BDNF-AS   \n",
       "3                risk      NaN         NaN        NaN       GWAS:WHRN   \n",
       "4                risk      NaN         NaN        NaN      GWAS:LIN7C   \n",
       "\n",
       "  evidence                                                tag  \n",
       "0      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "1      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "2      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "3      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "4      NaN  biomarker;assessed_biomarker_entity_id;assesse...  "
      ]
     },
     "execution_count": 264,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.insert(0, 'tmp_id', df.groupby(['biomarker', 'assessed_biomarker_entity_id', 'assessed_biomarker_entity', 'condition']).ngroup())\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>biomarker_id</th>\n",
       "      <th>biomarker</th>\n",
       "      <th>assessed_biomarker_entity</th>\n",
       "      <th>assessed_biomarker_entity_id</th>\n",
       "      <th>assessed_entity_type</th>\n",
       "      <th>condition</th>\n",
       "      <th>condition_id</th>\n",
       "      <th>exposure_agent</th>\n",
       "      <th>exposure_agent_id</th>\n",
       "      <th>best_biomarker_role</th>\n",
       "      <th>specimen</th>\n",
       "      <th>specimen_id</th>\n",
       "      <th>loinc_code</th>\n",
       "      <th>evidence_source</th>\n",
       "      <th>evidence</th>\n",
       "      <th>tag</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>533</td>\n",
       "      <td>presence of rs113988120 mutation in PAIP2B</td>\n",
       "      <td>PAIP2B</td>\n",
       "      <td>dbSNP:rs113988120</td>\n",
       "      <td>gene</td>\n",
       "      <td>pancreatic cancer</td>\n",
       "      <td>DOID:1793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GWAS:PAIP2B</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>176</td>\n",
       "      <td>presence of rs10736390 mutation in ACOT11</td>\n",
       "      <td>ACOT11</td>\n",
       "      <td>dbSNP:rs10736390</td>\n",
       "      <td>gene</td>\n",
       "      <td>pancreatic cancer</td>\n",
       "      <td>DOID:1793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GWAS:ACOT11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>193</td>\n",
       "      <td>presence of rs10767646 mutation in BDNF-AS</td>\n",
       "      <td>BDNF-AS</td>\n",
       "      <td>dbSNP:rs10767646</td>\n",
       "      <td>gene</td>\n",
       "      <td>pancreatic cancer</td>\n",
       "      <td>DOID:1793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GWAS:BDNF-AS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>218</td>\n",
       "      <td>presence of rs10817611 mutation in WHRN</td>\n",
       "      <td>WHRN</td>\n",
       "      <td>dbSNP:rs10817611</td>\n",
       "      <td>gene</td>\n",
       "      <td>pancreatic cancer</td>\n",
       "      <td>DOID:1793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GWAS:WHRN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>229</td>\n",
       "      <td>presence of rs10835188 mutation in LIN7C</td>\n",
       "      <td>LIN7C</td>\n",
       "      <td>dbSNP:rs10835188</td>\n",
       "      <td>gene</td>\n",
       "      <td>pancreatic cancer</td>\n",
       "      <td>DOID:1793</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>risk</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GWAS:LIN7C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   biomarker_id                                   biomarker  \\\n",
       "0           533  presence of rs113988120 mutation in PAIP2B   \n",
       "1           176   presence of rs10736390 mutation in ACOT11   \n",
       "2           193  presence of rs10767646 mutation in BDNF-AS   \n",
       "3           218     presence of rs10817611 mutation in WHRN   \n",
       "4           229    presence of rs10835188 mutation in LIN7C   \n",
       "\n",
       "  assessed_biomarker_entity assessed_biomarker_entity_id assessed_entity_type  \\\n",
       "0                    PAIP2B            dbSNP:rs113988120                 gene   \n",
       "1                    ACOT11             dbSNP:rs10736390                 gene   \n",
       "2                   BDNF-AS             dbSNP:rs10767646                 gene   \n",
       "3                      WHRN             dbSNP:rs10817611                 gene   \n",
       "4                     LIN7C             dbSNP:rs10835188                 gene   \n",
       "\n",
       "           condition condition_id exposure_agent exposure_agent_id  \\\n",
       "0  pancreatic cancer    DOID:1793            NaN               NaN   \n",
       "1  pancreatic cancer    DOID:1793            NaN               NaN   \n",
       "2  pancreatic cancer    DOID:1793            NaN               NaN   \n",
       "3  pancreatic cancer    DOID:1793            NaN               NaN   \n",
       "4  pancreatic cancer    DOID:1793            NaN               NaN   \n",
       "\n",
       "  best_biomarker_role specimen specimen_id loinc_code evidence_source  \\\n",
       "0                risk      NaN         NaN        NaN     GWAS:PAIP2B   \n",
       "1                risk      NaN         NaN        NaN     GWAS:ACOT11   \n",
       "2                risk      NaN         NaN        NaN    GWAS:BDNF-AS   \n",
       "3                risk      NaN         NaN        NaN       GWAS:WHRN   \n",
       "4                risk      NaN         NaN        NaN      GWAS:LIN7C   \n",
       "\n",
       "  evidence                                                tag  \n",
       "0      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "1      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "2      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "3      NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "4      NaN  biomarker;assessed_biomarker_entity_id;assesse...  "
      ]
     },
     "execution_count": 269,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# drop biomarker_id column and rename tmp_id to biomarker_id\n",
    "df = df.drop('biomarker_id', axis = 1)\n",
    "df = df.rename(columns = {'tmp_id': 'biomarker_id'})\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 270,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save cleaned data \n",
    "df.to_csv('../home/data/cleaned_data/GlyGen/gwas.tsv', sep = '\\t', index = False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
